{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import subprocess\n",
    "import pandas as pd\n",
    "import pybedtools\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get ADPD SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['chr', 'pos', 'r2_with_ld_tag', 'ld_tag_chr', 'ld_tag_pos',\n",
       "       'source_gwas', 'source', 'snp_id', 'locus_num', 'number_ld_buddies',\n",
       "       'ld_tag_locus', 'rsid', 'chrom_hg19', 'snp_pos_hg19', 'file',\n",
       "       'effect_allele', 'noneffect_allele', 'direction', 'pvalue', 'has_coloc',\n",
       "       'direct_atac_overlap_narrow_tissue_regions',\n",
       "       'containing_atac_tissues_narrow_tissue_regions',\n",
       "       'nearest_atac_tissue_narrow_tissue_regions',\n",
       "       'start_narrow_tissue_regions', 'end_narrow_tissue_regions',\n",
       "       'dist_narrow_tissue_regions',\n",
       "       'direct_atac_overlap_broad_tissue_regions',\n",
       "       'containing_atac_tissues_broad_tissue_regions',\n",
       "       'nearest_atac_tissue_broad_tissue_regions',\n",
       "       'start_broad_tissue_regions', 'end_broad_tissue_regions',\n",
       "       'dist_broad_tissue_regions', 'direct_atac_overlap_single_cell',\n",
       "       'containing_atac_tissues_single_cell',\n",
       "       'nearest_atac_tissue_single_cell', 'start_single_cell',\n",
       "       'end_single_cell', 'dist_single_cell'],\n",
       "      dtype='object')"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "adpd_snps = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/snps_final/191121_ld_buddies_table_stage3.tsv', sep='\\t')\n",
    "display(adpd_snps.columns)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make hg19 SNP BED files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom_hg19</th>\n",
       "      <th>start</th>\n",
       "      <th>snp_pos_hg19</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>2334</td>\n",
       "      <td>chr1</td>\n",
       "      <td>40287822</td>\n",
       "      <td>40287823</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2335</td>\n",
       "      <td>chr1</td>\n",
       "      <td>40287916</td>\n",
       "      <td>40287917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2336</td>\n",
       "      <td>chr1</td>\n",
       "      <td>40287931</td>\n",
       "      <td>40287932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2337</td>\n",
       "      <td>chr1</td>\n",
       "      <td>40288463</td>\n",
       "      <td>40288464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2338</td>\n",
       "      <td>chr1</td>\n",
       "      <td>40290896</td>\n",
       "      <td>40290897</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13768</td>\n",
       "      <td>chr9</td>\n",
       "      <td>34036528</td>\n",
       "      <td>34036529</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13769</td>\n",
       "      <td>chr9</td>\n",
       "      <td>34039704</td>\n",
       "      <td>34039705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13770</td>\n",
       "      <td>chr9</td>\n",
       "      <td>34040773</td>\n",
       "      <td>34040774</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13771</td>\n",
       "      <td>chr9</td>\n",
       "      <td>34046390</td>\n",
       "      <td>34046391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13772</td>\n",
       "      <td>chr9</td>\n",
       "      <td>34046807</td>\n",
       "      <td>34046808</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9538 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom_hg19     start  snp_pos_hg19\n",
       "2334        chr1  40287822      40287823\n",
       "2335        chr1  40287916      40287917\n",
       "2336        chr1  40287931      40287932\n",
       "2337        chr1  40288463      40288464\n",
       "2338        chr1  40290896      40290897\n",
       "...          ...       ...           ...\n",
       "13768       chr9  34036528      34036529\n",
       "13769       chr9  34039704      34039705\n",
       "13770       chr9  34040773      34040774\n",
       "13771       chr9  34046390      34046391\n",
       "13772       chr9  34046807      34046808\n",
       "\n",
       "[9538 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "snps_bed_hg19 = adpd_snps[['chrom_hg19', 'snp_pos_hg19']]\n",
    "snps_bed_hg19 = snps_bed_hg19.loc[snps_bed_hg19['chrom_hg19'].notnull()]\n",
    "snps_bed_hg19['chrom_hg19'] = snps_bed_hg19['chrom_hg19'].astype(int).astype(str)\n",
    "snps_bed_hg19['snp_pos_hg19'] = snps_bed_hg19['snp_pos_hg19'].astype(int)\n",
    "snps_bed_hg19['chrom_hg19'] = 'chr' + snps_bed_hg19['chrom_hg19']\n",
    "snps_bed_hg19['start'] = snps_bed_hg19['snp_pos_hg19'] - 1\n",
    "snps_bed_hg19 = snps_bed_hg19[['chrom_hg19', 'start', 'snp_pos_hg19']]\n",
    "snps_bed_hg19.sort_values(by=['chrom_hg19', 'start', 'snp_pos_hg19'], inplace=True)\n",
    "snps_bed_hg19.drop_duplicates(inplace=True)\n",
    "display(snps_bed_hg19)\n",
    "snps_bed_hg19.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/hg19_snps.bed', sep='\\t', index=False, header=False)\n",
    "\n",
    "nochr_snps_bed_hg19 = adpd_snps[['chrom_hg19', 'snp_pos_hg19']]\n",
    "nochr_snps_bed_hg19 = nochr_snps_bed_hg19.loc[nochr_snps_bed_hg19['chrom_hg19'].notnull()]\n",
    "nochr_snps_bed_hg19['chrom_hg19'] = nochr_snps_bed_hg19['chrom_hg19'].astype(int).astype(str)\n",
    "nochr_snps_bed_hg19['snp_pos_hg19'] = nochr_snps_bed_hg19['snp_pos_hg19'].astype(int)\n",
    "nochr_snps_bed_hg19['start'] = nochr_snps_bed_hg19['snp_pos_hg19'] - 1\n",
    "nochr_snps_bed_hg19 = nochr_snps_bed_hg19[['chrom_hg19', 'start', 'snp_pos_hg19']]\n",
    "nochr_snps_bed_hg19.sort_values(by=['chrom_hg19', 'start', 'snp_pos_hg19'], inplace=True)\n",
    "nochr_snps_bed_hg19.drop_duplicates(inplace=True)\n",
    "nochr_snps_bed_hg19.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/nochr_hg19_snps.bed', sep='\\t', index=False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get hg38 Coordinates for 1KG intersected SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>score</th>\n",
       "      <th>qc</th>\n",
       "      <th>info</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1230</td>\n",
       "      <td>.;esv3626682</td>\n",
       "      <td>T</td>\n",
       "      <td>&lt;CN0&gt;,&lt;CN2&gt;</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0,1;AF=0,0.000199681;AN=5008;CS=DUP_gs;END=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9680</td>\n",
       "      <td>DUP_uwash_chr5_60140563_60179087</td>\n",
       "      <td>A</td>\n",
       "      <td>&lt;CN0&gt;</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=0;AF=0;AN=5008;CS=DUP_uwash;END=60179087;NS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>esv3585792</td>\n",
       "      <td>G</td>\n",
       "      <td>&lt;CN0&gt;</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.000199681;AN=5008;CIEND=0,500;CIPOS=...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>45</td>\n",
       "      <td>esv3585794</td>\n",
       "      <td>C</td>\n",
       "      <td>&lt;CN0&gt;</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.000199681;AN=5008;CIEND=-23,24;CIPOS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64</td>\n",
       "      <td>esv3587615</td>\n",
       "      <td>T</td>\n",
       "      <td>&lt;CN2&gt;</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1;AF=0.000199681;AN=5008;CIEND=-150,150;CIP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8127</td>\n",
       "      <td>rs9984928</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1660;AF=0.33147;AN=5008;NS=2504;DP=16963;EA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7588</td>\n",
       "      <td>rs9989786</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=3719;AF=0.742612;AN=5008;NS=2504;DP=19926;E...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8852</td>\n",
       "      <td>rs9994159</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=599;AF=0.119609;AN=5008;NS=2504;DP=20517;EA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9357</td>\n",
       "      <td>rs9995651</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=310;AF=0.061901;AN=5008;NS=2504;DP=13879;EA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9225</td>\n",
       "      <td>rs9999231</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=1833;AF=0.366014;AN=5008;NS=2504;DP=14651;E...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9757 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  rsid ref          alt  score    qc  \\\n",
       "1230                      .;esv3626682   T  <CN0>,<CN2>    100  PASS   \n",
       "9680  DUP_uwash_chr5_60140563_60179087   A        <CN0>    100  PASS   \n",
       "0                           esv3585792   G        <CN0>    100  PASS   \n",
       "45                          esv3585794   C        <CN0>    100  PASS   \n",
       "64                          esv3587615   T        <CN2>    100  PASS   \n",
       "...                                ...  ..          ...    ...   ...   \n",
       "8127                         rs9984928   G            A    100  PASS   \n",
       "7588                         rs9989786   A            G    100  PASS   \n",
       "8852                         rs9994159   C            T    100  PASS   \n",
       "9357                         rs9995651   C            G    100  PASS   \n",
       "9225                         rs9999231   T            C    100  PASS   \n",
       "\n",
       "                                                   info  \n",
       "1230  AC=0,1;AF=0,0.000199681;AN=5008;CS=DUP_gs;END=...  \n",
       "9680  AC=0;AF=0;AN=5008;CS=DUP_uwash;END=60179087;NS...  \n",
       "0     AC=1;AF=0.000199681;AN=5008;CIEND=0,500;CIPOS=...  \n",
       "45    AC=1;AF=0.000199681;AN=5008;CIEND=-23,24;CIPOS...  \n",
       "64    AC=1;AF=0.000199681;AN=5008;CIEND=-150,150;CIP...  \n",
       "...                                                 ...  \n",
       "8127  AC=1660;AF=0.33147;AN=5008;NS=2504;DP=16963;EA...  \n",
       "7588  AC=3719;AF=0.742612;AN=5008;NS=2504;DP=19926;E...  \n",
       "8852  AC=599;AF=0.119609;AN=5008;NS=2504;DP=20517;EA...  \n",
       "9357  AC=310;AF=0.061901;AN=5008;NS=2504;DP=13879;EA...  \n",
       "9225  AC=1833;AF=0.366014;AN=5008;NS=2504;DP=14651;E...  \n",
       "\n",
       "[9757 rows x 6 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hg19_vcf = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/1KG_ADPD_hg19_snps.vcf', header=None, sep='\\t')\n",
    "hg19_vcf = hg19_vcf[[2, 3, 4, 5, 6, 7]]\n",
    "hg19_vcf.columns = ['rsid', 'ref', 'alt', 'score', 'qc', 'info']\n",
    "hg19_vcf.sort_values(by=['rsid', 'ref'], inplace=True)\n",
    "hg19_vcf.drop_duplicates(inplace=True)\n",
    "hg19_vcf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>2334</td>\n",
       "      <td>1</td>\n",
       "      <td>39822151</td>\n",
       "      <td>rs34640847</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2335</td>\n",
       "      <td>1</td>\n",
       "      <td>39822245</td>\n",
       "      <td>rs36015266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2336</td>\n",
       "      <td>1</td>\n",
       "      <td>39822260</td>\n",
       "      <td>rs61779808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2337</td>\n",
       "      <td>1</td>\n",
       "      <td>39822792</td>\n",
       "      <td>rs61779809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2338</td>\n",
       "      <td>1</td>\n",
       "      <td>39825225</td>\n",
       "      <td>rs72666941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10156</td>\n",
       "      <td>21</td>\n",
       "      <td>37517450</td>\n",
       "      <td>rs2236688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10157</td>\n",
       "      <td>21</td>\n",
       "      <td>37517672</td>\n",
       "      <td>rs2835776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10158</td>\n",
       "      <td>21</td>\n",
       "      <td>37519947</td>\n",
       "      <td>rs11701836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10159</td>\n",
       "      <td>21</td>\n",
       "      <td>37527140</td>\n",
       "      <td>rs7280075</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10272</td>\n",
       "      <td>22</td>\n",
       "      <td>46324889</td>\n",
       "      <td>rs75718201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9707 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr       pos        rsid\n",
       "2334     1  39822151  rs34640847\n",
       "2335     1  39822245  rs36015266\n",
       "2336     1  39822260  rs61779808\n",
       "2337     1  39822792  rs61779809\n",
       "2338     1  39825225  rs72666941\n",
       "...    ...       ...         ...\n",
       "10156   21  37517450   rs2236688\n",
       "10157   21  37517672   rs2835776\n",
       "10158   21  37519947  rs11701836\n",
       "10159   21  37527140   rs7280075\n",
       "10272   22  46324889  rs75718201\n",
       "\n",
       "[9707 rows x 3 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adpd_snps = adpd_snps[['chr', 'pos', 'rsid']]\n",
    "adpd_snps.sort_values(by=['chr', 'pos'], inplace=True)\n",
    "adpd_snps.drop_duplicates(inplace=True)\n",
    "adpd_snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged = adpd_snps.merge(hg19_vcf, on='rsid')\n",
    "merged['chr'] = 'chr' + merged['chr'].astype('str')\n",
    "merged['start'] = merged['pos'] - 1\n",
    "merged = merged.loc[merged['alt'].apply(lambda x : len(x) < 2)]\n",
    "merged = merged.loc[merged['ref'].apply(lambda x : len(x) < 2)]\n",
    "# hg38_bed = merged[['chr', 'start', 'pos', 'rsid', 'ref', 'alt', 'info']]\n",
    "# hg38_bed['info'] = hg38_bed['info'].apply(lambda x : x.split('EUR_AF=')[1].split(';')[0])\n",
    "# hg38_bed.sort_values(by=['chr', 'pos'], inplace=True)\n",
    "# hg38_bed.drop_duplicates(inplace=True)\n",
    "# hg38_bed.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/hg38_snps_1KG_ADPD.bed', header=False, index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make New VCF file from merged file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>pos</th>\n",
       "      <th>rsid</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>score</th>\n",
       "      <th>qc</th>\n",
       "      <th>info</th>\n",
       "      <th>format</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>39822151</td>\n",
       "      <td>rs34640847</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=595;AF=0.11881;AN=5008;NS=2504;DP=18110;EAS...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>39822245</td>\n",
       "      <td>rs36015266</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=595;AF=0.11881;AN=5008;NS=2504;DP=19296;EAS...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>39822260</td>\n",
       "      <td>rs61779808</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=493;AF=0.0984425;AN=5008;NS=2504;DP=19241;E...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>39822792</td>\n",
       "      <td>rs61779809</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=595;AF=0.11881;AN=5008;NS=2504;DP=19613;EAS...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr1</td>\n",
       "      <td>39825225</td>\n",
       "      <td>rs72666941</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=477;AF=0.0952476;AN=5008;NS=2504;DP=20561;E...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3305</td>\n",
       "      <td>chr9</td>\n",
       "      <td>34036531</td>\n",
       "      <td>rs1629188</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=3379;AF=0.67472;AN=5008;NS=2504;DP=18771;EA...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3306</td>\n",
       "      <td>chr9</td>\n",
       "      <td>34039707</td>\n",
       "      <td>rs9696412</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=3379;AF=0.67472;AN=5008;NS=2504;DP=13159;EA...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3307</td>\n",
       "      <td>chr9</td>\n",
       "      <td>34040776</td>\n",
       "      <td>rs1543605</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=3379;AF=0.67472;AN=5008;NS=2504;DP=21998;EA...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3308</td>\n",
       "      <td>chr9</td>\n",
       "      <td>34046393</td>\n",
       "      <td>rs6476434</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=3379;AF=0.67472;AN=5008;NS=2504;DP=15262;EA...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3309</td>\n",
       "      <td>chr9</td>\n",
       "      <td>34046810</td>\n",
       "      <td>rs7019833</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>100</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AC=3379;AF=0.67472;AN=5008;NS=2504;DP=19550;EA...</td>\n",
       "      <td>GT</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9354 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr       pos        rsid ref alt  score    qc  \\\n",
       "0     chr1  39822151  rs34640847   G   A    100  PASS   \n",
       "1     chr1  39822245  rs36015266   C   T    100  PASS   \n",
       "2     chr1  39822260  rs61779808   T   A    100  PASS   \n",
       "3     chr1  39822792  rs61779809   C   T    100  PASS   \n",
       "4     chr1  39825225  rs72666941   C   A    100  PASS   \n",
       "...    ...       ...         ...  ..  ..    ...   ...   \n",
       "3305  chr9  34036531   rs1629188   G   C    100  PASS   \n",
       "3306  chr9  34039707   rs9696412   A   G    100  PASS   \n",
       "3307  chr9  34040776   rs1543605   C   G    100  PASS   \n",
       "3308  chr9  34046393   rs6476434   C   T    100  PASS   \n",
       "3309  chr9  34046810   rs7019833   A   C    100  PASS   \n",
       "\n",
       "                                                   info format  \n",
       "0     AC=595;AF=0.11881;AN=5008;NS=2504;DP=18110;EAS...     GT  \n",
       "1     AC=595;AF=0.11881;AN=5008;NS=2504;DP=19296;EAS...     GT  \n",
       "2     AC=493;AF=0.0984425;AN=5008;NS=2504;DP=19241;E...     GT  \n",
       "3     AC=595;AF=0.11881;AN=5008;NS=2504;DP=19613;EAS...     GT  \n",
       "4     AC=477;AF=0.0952476;AN=5008;NS=2504;DP=20561;E...     GT  \n",
       "...                                                 ...    ...  \n",
       "3305  AC=3379;AF=0.67472;AN=5008;NS=2504;DP=18771;EA...     GT  \n",
       "3306  AC=3379;AF=0.67472;AN=5008;NS=2504;DP=13159;EA...     GT  \n",
       "3307  AC=3379;AF=0.67472;AN=5008;NS=2504;DP=21998;EA...     GT  \n",
       "3308  AC=3379;AF=0.67472;AN=5008;NS=2504;DP=15262;EA...     GT  \n",
       "3309  AC=3379;AF=0.67472;AN=5008;NS=2504;DP=19550;EA...     GT  \n",
       "\n",
       "[9354 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "hg38_vcf = merged[['chr', 'pos', 'rsid', 'ref', 'alt', 'score', 'qc', 'info']]\n",
    "hg38_vcf['format'] = ['GT' for i in range(len(hg38_vcf))]\n",
    "hg38_vcf = hg38_vcf.sort_values(by=['chr', 'pos'])\n",
    "hg38_vcf.drop_duplicates(inplace=True)\n",
    "display(hg38_vcf)\n",
    "hg38_vcf.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/hg38_snps_1KG_ADPD.vcf', header=False, index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get ADPD Metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OldName</th>\n",
       "      <th>NewName</th>\n",
       "      <th>HarmonizedName</th>\n",
       "      <th>Bam</th>\n",
       "      <th>Contrast</th>\n",
       "      <th>Type</th>\n",
       "      <th>Group</th>\n",
       "      <th>Region</th>\n",
       "      <th>RegionMod</th>\n",
       "      <th>PatientID</th>\n",
       "      <th>...</th>\n",
       "      <th>xxx.CONTRIBUTING_NP_DX</th>\n",
       "      <th>xxx.Interval between death and last MMSE</th>\n",
       "      <th>xxx.ch_lastCasiScore</th>\n",
       "      <th>xxx.ch_lastCasiDate</th>\n",
       "      <th>xxx.micro_AmyloidAngiopathyOccipitalLobe_ID</th>\n",
       "      <th>xxx.GE_atherosclerosis_ID</th>\n",
       "      <th>xxx.calc_A</th>\n",
       "      <th>xxx.calc_B</th>\n",
       "      <th>xxx.C</th>\n",
       "      <th>xxx.CognitiveStatus</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>PD_00_38_CTRL_CAUD_X014_S01_L001_B1_T1_P025</td>\n",
       "      <td>PD_00_38_CTRL_CAUD_X014_S01_L001_B1_T1_P025</td>\n",
       "      <td>CTRL_CAUD_PD_00x38xx_X014_S01_L001_B1_T1_P025</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_CAUD</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>00_38</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>PD_00_38_CTRL_CAUD_X014_S01_L002_B1_T2_P028</td>\n",
       "      <td>PD_00_38_CTRL_CAUD_X014_S01_L002_B1_T2_P028</td>\n",
       "      <td>CTRL_CAUD_PD_00x38xx_X014_S01_L002_B1_T2_P028</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_CAUD</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>CAUD</td>\n",
       "      <td>00_38</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>PD_00_38_CTRL_HIPP_X002_S11_L045_B1_T1_P002</td>\n",
       "      <td>PD_00_38_CTRL_HIPP_X002_S11_L045_B1_T1_P002</td>\n",
       "      <td>CTRL_HIPP_PD_00x38xx_X002_S11_L045_B1_T1_P002</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_HIPP</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>HIPP</td>\n",
       "      <td>HIPP</td>\n",
       "      <td>00_38</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>PD_00_38_CTRL_HIPP_X002_S11_L046_B1_T2_P003</td>\n",
       "      <td>PD_00_38_CTRL_HIPP_X002_S11_L046_B1_T2_P003</td>\n",
       "      <td>CTRL_HIPP_PD_00x38xx_X002_S11_L046_B1_T2_P003</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_HIPP</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>HIPP</td>\n",
       "      <td>HIPP</td>\n",
       "      <td>00_38</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>PD_00_38_CTRL_MDFG_X007_S04_L055_B1_T1_P014</td>\n",
       "      <td>PD_00_38_CTRL_MDFG_X007_S04_L055_B1_T1_P014</td>\n",
       "      <td>CTRL_MDFG_PD_00x38xx_X007_S04_L055_B1_T1_P014</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_MDFG</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>MDFG</td>\n",
       "      <td>00_38</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>263</td>\n",
       "      <td>PD_16_32_CTRL_MDTG_X003_S04_L008_B1_T2_P003</td>\n",
       "      <td>PD_16_32_CTRL_MDTG_X003_S04_L008_B1_T2_P003</td>\n",
       "      <td>CTRL_SMTG_PD_16x32xx_X003_S04_L008_B1_T2_P003</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_MDTG</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>MDTG</td>\n",
       "      <td>SMTG</td>\n",
       "      <td>16_32</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>264</td>\n",
       "      <td>PD_16_32_CTRL_PTMN_X019_S08_L039_B1_T1_P033</td>\n",
       "      <td>PD_16_32_CTRL_PTMN_X019_S08_L039_B1_T1_P033</td>\n",
       "      <td>CTRL_PTMN_PD_16x32xx_X019_S08_L039_B1_T1_P033</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_PTMN</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>16_32</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>265</td>\n",
       "      <td>PD_16_32_CTRL_PTMN_X019_S08_L040_B1_T2_P030</td>\n",
       "      <td>PD_16_32_CTRL_PTMN_X019_S08_L040_B1_T2_P030</td>\n",
       "      <td>CTRL_PTMN_PD_16x32xx_X019_S08_L040_B1_T2_P030</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_PTMN</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>PTMN</td>\n",
       "      <td>16_32</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>266</td>\n",
       "      <td>PD_16_32_CTRL_SUNI_X007_S12_L071_B1_T1_P014</td>\n",
       "      <td>PD_16_32_CTRL_SUNI_X007_S12_L071_B1_T1_P014</td>\n",
       "      <td>CTRL_SUNI_PD_16x32xx_X007_S12_L071_B1_T1_P014</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_SUNI</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>SUNI</td>\n",
       "      <td>SUNI</td>\n",
       "      <td>16_32</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>267</td>\n",
       "      <td>PD_16_32_CTRL_SUNI_X007_S12_L072_B1_T2_P012</td>\n",
       "      <td>PD_16_32_CTRL_SUNI_X007_S12_L072_B1_T2_P012</td>\n",
       "      <td>CTRL_SUNI_PD_16x32xx_X007_S12_L072_B1_T2_P012</td>\n",
       "      <td>/oak/stanford/groups/howchang/users/mcorces/PD...</td>\n",
       "      <td>CTRL_SUNI</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>CTRL</td>\n",
       "      <td>SUNI</td>\n",
       "      <td>SUNI</td>\n",
       "      <td>16_32</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>268 rows × 108 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         OldName  \\\n",
       "0    PD_00_38_CTRL_CAUD_X014_S01_L001_B1_T1_P025   \n",
       "1    PD_00_38_CTRL_CAUD_X014_S01_L002_B1_T2_P028   \n",
       "2    PD_00_38_CTRL_HIPP_X002_S11_L045_B1_T1_P002   \n",
       "3    PD_00_38_CTRL_HIPP_X002_S11_L046_B1_T2_P003   \n",
       "4    PD_00_38_CTRL_MDFG_X007_S04_L055_B1_T1_P014   \n",
       "..                                           ...   \n",
       "263  PD_16_32_CTRL_MDTG_X003_S04_L008_B1_T2_P003   \n",
       "264  PD_16_32_CTRL_PTMN_X019_S08_L039_B1_T1_P033   \n",
       "265  PD_16_32_CTRL_PTMN_X019_S08_L040_B1_T2_P030   \n",
       "266  PD_16_32_CTRL_SUNI_X007_S12_L071_B1_T1_P014   \n",
       "267  PD_16_32_CTRL_SUNI_X007_S12_L072_B1_T2_P012   \n",
       "\n",
       "                                         NewName  \\\n",
       "0    PD_00_38_CTRL_CAUD_X014_S01_L001_B1_T1_P025   \n",
       "1    PD_00_38_CTRL_CAUD_X014_S01_L002_B1_T2_P028   \n",
       "2    PD_00_38_CTRL_HIPP_X002_S11_L045_B1_T1_P002   \n",
       "3    PD_00_38_CTRL_HIPP_X002_S11_L046_B1_T2_P003   \n",
       "4    PD_00_38_CTRL_MDFG_X007_S04_L055_B1_T1_P014   \n",
       "..                                           ...   \n",
       "263  PD_16_32_CTRL_MDTG_X003_S04_L008_B1_T2_P003   \n",
       "264  PD_16_32_CTRL_PTMN_X019_S08_L039_B1_T1_P033   \n",
       "265  PD_16_32_CTRL_PTMN_X019_S08_L040_B1_T2_P030   \n",
       "266  PD_16_32_CTRL_SUNI_X007_S12_L071_B1_T1_P014   \n",
       "267  PD_16_32_CTRL_SUNI_X007_S12_L072_B1_T2_P012   \n",
       "\n",
       "                                    HarmonizedName  \\\n",
       "0    CTRL_CAUD_PD_00x38xx_X014_S01_L001_B1_T1_P025   \n",
       "1    CTRL_CAUD_PD_00x38xx_X014_S01_L002_B1_T2_P028   \n",
       "2    CTRL_HIPP_PD_00x38xx_X002_S11_L045_B1_T1_P002   \n",
       "3    CTRL_HIPP_PD_00x38xx_X002_S11_L046_B1_T2_P003   \n",
       "4    CTRL_MDFG_PD_00x38xx_X007_S04_L055_B1_T1_P014   \n",
       "..                                             ...   \n",
       "263  CTRL_SMTG_PD_16x32xx_X003_S04_L008_B1_T2_P003   \n",
       "264  CTRL_PTMN_PD_16x32xx_X019_S08_L039_B1_T1_P033   \n",
       "265  CTRL_PTMN_PD_16x32xx_X019_S08_L040_B1_T2_P030   \n",
       "266  CTRL_SUNI_PD_16x32xx_X007_S12_L071_B1_T1_P014   \n",
       "267  CTRL_SUNI_PD_16x32xx_X007_S12_L072_B1_T2_P012   \n",
       "\n",
       "                                                   Bam   Contrast  Type Group  \\\n",
       "0    /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_CAUD  CTRL  CTRL   \n",
       "1    /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_CAUD  CTRL  CTRL   \n",
       "2    /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_HIPP  CTRL  CTRL   \n",
       "3    /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_HIPP  CTRL  CTRL   \n",
       "4    /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_MDFG  CTRL  CTRL   \n",
       "..                                                 ...        ...   ...   ...   \n",
       "263  /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_MDTG  CTRL  CTRL   \n",
       "264  /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_PTMN  CTRL  CTRL   \n",
       "265  /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_PTMN  CTRL  CTRL   \n",
       "266  /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_SUNI  CTRL  CTRL   \n",
       "267  /oak/stanford/groups/howchang/users/mcorces/PD...  CTRL_SUNI  CTRL  CTRL   \n",
       "\n",
       "    Region RegionMod PatientID  ... xxx.CONTRIBUTING_NP_DX  \\\n",
       "0     CAUD      CAUD     00_38  ...                    NaN   \n",
       "1     CAUD      CAUD     00_38  ...                    NaN   \n",
       "2     HIPP      HIPP     00_38  ...                    NaN   \n",
       "3     HIPP      HIPP     00_38  ...                    NaN   \n",
       "4     MDFG      MDFG     00_38  ...                    NaN   \n",
       "..     ...       ...       ...  ...                    ...   \n",
       "263   MDTG      SMTG     16_32  ...                    NaN   \n",
       "264   PTMN      PTMN     16_32  ...                    NaN   \n",
       "265   PTMN      PTMN     16_32  ...                    NaN   \n",
       "266   SUNI      SUNI     16_32  ...                    NaN   \n",
       "267   SUNI      SUNI     16_32  ...                    NaN   \n",
       "\n",
       "    xxx.Interval between death and last MMSE xxx.ch_lastCasiScore  \\\n",
       "0                                        NaN                  NaN   \n",
       "1                                        NaN                  NaN   \n",
       "2                                        NaN                  NaN   \n",
       "3                                        NaN                  NaN   \n",
       "4                                        NaN                  NaN   \n",
       "..                                       ...                  ...   \n",
       "263                                      NaN                  NaN   \n",
       "264                                      NaN                  NaN   \n",
       "265                                      NaN                  NaN   \n",
       "266                                      NaN                  NaN   \n",
       "267                                      NaN                  NaN   \n",
       "\n",
       "    xxx.ch_lastCasiDate xxx.micro_AmyloidAngiopathyOccipitalLobe_ID  \\\n",
       "0                   NaN                                         NaN   \n",
       "1                   NaN                                         NaN   \n",
       "2                   NaN                                         NaN   \n",
       "3                   NaN                                         NaN   \n",
       "4                   NaN                                         NaN   \n",
       "..                  ...                                         ...   \n",
       "263                 NaN                                         NaN   \n",
       "264                 NaN                                         NaN   \n",
       "265                 NaN                                         NaN   \n",
       "266                 NaN                                         NaN   \n",
       "267                 NaN                                         NaN   \n",
       "\n",
       "    xxx.GE_atherosclerosis_ID xxx.calc_A xxx.calc_B xxx.C xxx.CognitiveStatus  \n",
       "0                         NaN        NaN        NaN   NaN                 NaN  \n",
       "1                         NaN        NaN        NaN   NaN                 NaN  \n",
       "2                         NaN        NaN        NaN   NaN                 NaN  \n",
       "3                         NaN        NaN        NaN   NaN                 NaN  \n",
       "4                         NaN        NaN        NaN   NaN                 NaN  \n",
       "..                        ...        ...        ...   ...                 ...  \n",
       "263                       NaN        NaN        NaN   NaN                 NaN  \n",
       "264                       NaN        NaN        NaN   NaN                 NaN  \n",
       "265                       NaN        NaN        NaN   NaN                 NaN  \n",
       "266                       NaN        NaN        NaN   NaN                 NaN  \n",
       "267                       NaN        NaN        NaN   NaN                 NaN  \n",
       "\n",
       "[268 rows x 108 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "metadata = pd.read_excel('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/190215_Brain-ControlsOnly_Metadata_Merged.xlsx')\n",
    "display(metadata)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get Bams for each brain region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['CAUD', 'HIPP', 'MDFG', 'MDTG', 'PTMN', 'SUNI', 'SMTG', 'PARL'],\n",
       "      dtype=object)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(metadata['Region'].unique())\n",
    "regions = list(metadata['Region'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAUD\n",
      "HIPP\n",
      "MDFG\n",
      "MDTG\n",
      "PTMN\n",
      "SUNI\n",
      "SMTG\n",
      "PARL\n"
     ]
    }
   ],
   "source": [
    "with open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/ctrl_bams.txt') as infile:\n",
    "    ctrl_bams = infile.readlines()\n",
    "multi_bam_regs = []\n",
    "for reg in regions:\n",
    "    print(reg)\n",
    "    region_bams = []\n",
    "    for bam in ctrl_bams:\n",
    "        if reg in bam:\n",
    "            region_bams.append(bam)\n",
    "    if len(region_bams) > 0:\n",
    "        multi_bam_regs.append(reg)\n",
    "        region_bams.sort()\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg)\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/' + reg):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/' + reg)\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/' + reg):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/' + reg)\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/logs/' + reg):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/logs/' + reg)\n",
    "        with open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/bam_lists/' + reg + '_ctrl_bams.txt', 'w') as outfile:\n",
    "            for bam in region_bams:\n",
    "                outfile.write(bam)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Make Input VCF files for each region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for reg in multi_bam_regs:\n",
    "    print(reg)\n",
    "    with open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/bam_lists/' + reg + '_ctrl_bams.txt') as infile:\n",
    "        reg_bams = [i.strip() for i in infile.readlines()]\n",
    "    reg_vcf = hg38_vcf.copy(deep=True)\n",
    "    for bam in reg_bams:\n",
    "        bam_name = bam.split('/')[10]\n",
    "        #print(bam_name)\n",
    "        reg_vcf[bam_name] = ['./.' for i in range(len(reg_vcf))]\n",
    "    #reg_vcf.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/initial.vcf', header=False, index=False, sep='\\t')\n",
    "    bgzip_cmd = 'bgzip /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/initial.vcf'\n",
    "    #! {bgzip_cmd}\n",
    "    tabix_cmd = 'tabix -p vcf /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/initial.vcf.gz'\n",
    "    #! {tabix_cmd}\n",
    "    asvcf_cmd_1 = 'echo sbatch --export=ALL -n 1 -t 1-0 -p akundaje --mail-type=ALL -J ' \\\n",
    "                + reg + ' -o /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/logs/' \\\n",
    "                + reg + '/asvcf.o -e /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/logs/' \\\n",
    "                + reg + '/asvcf.e /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/' \\\n",
    "                + reg + '/asvcf.sh >> /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/run_asvcf.sh'\n",
    "    asvcf_cmd_2 = 'echo \\'#!/bin/bash\\' > /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/' \\\n",
    "                + reg + '/asvcf.sh'\n",
    "    asvcf_cmd_3 = 'echo /home/users/soumyak/rasqual/src/ASVCF/createASVCF.sh paired_end ' \\\n",
    "                + '/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/bam_lists/' \\\n",
    "                + reg + '_ctrl_bams.txt /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' \\\n",
    "                + reg + '/initial.vcf.gz /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' \\\n",
    "                + reg + '/asvcf.gz atac >> /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/' \\\n",
    "                + reg + '/asvcf.sh'\n",
    "    print(asvcf_cmd_1)\n",
    "    print(asvcf_cmd_2)\n",
    "    print(asvcf_cmd_3)\n",
    "    ! {asvcf_cmd_1}\n",
    "    ! {asvcf_cmd_2}\n",
    "    ! {asvcf_cmd_3}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load IDR Counts Matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ADAD_CAUD_00_0281</th>\n",
       "      <th>ADAD_CAUD_00_0387</th>\n",
       "      <th>ADAD_CAUD_01_0164</th>\n",
       "      <th>ADAD_CAUD_01_1400</th>\n",
       "      <th>ADAD_CAUD_06_0194</th>\n",
       "      <th>ADAD_CAUD_06_1486</th>\n",
       "      <th>ADAD_CAUD_07_0787</th>\n",
       "      <th>...</th>\n",
       "      <th>LRRK_MDTG_01_39</th>\n",
       "      <th>LRRK_MDTG_04_10</th>\n",
       "      <th>LRRK_MDTG_10_37</th>\n",
       "      <th>LRRK_MDTG_13_60</th>\n",
       "      <th>LRRK_PTMN_01_39</th>\n",
       "      <th>LRRK_PTMN_04_10</th>\n",
       "      <th>LRRK_PTMN_10_37</th>\n",
       "      <th>LRRK_PTMN_13_60</th>\n",
       "      <th>LRRK_SUNI_04_10</th>\n",
       "      <th>LRRK_SUNI_10_37</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10015</td>\n",
       "      <td>10231</td>\n",
       "      <td>12</td>\n",
       "      <td>16</td>\n",
       "      <td>22</td>\n",
       "      <td>12</td>\n",
       "      <td>14</td>\n",
       "      <td>20</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>9</td>\n",
       "      <td>22</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>181363</td>\n",
       "      <td>181563</td>\n",
       "      <td>1</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr1</td>\n",
       "      <td>183716</td>\n",
       "      <td>183916</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>10</td>\n",
       "      <td>19</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>10</td>\n",
       "      <td>21</td>\n",
       "      <td>3</td>\n",
       "      <td>16</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>20</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr1</td>\n",
       "      <td>184083</td>\n",
       "      <td>184283</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>20</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>20</td>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "      <td>3</td>\n",
       "      <td>22</td>\n",
       "      <td>18</td>\n",
       "      <td>11</td>\n",
       "      <td>26</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr1</td>\n",
       "      <td>184370</td>\n",
       "      <td>184570</td>\n",
       "      <td>7</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>6</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>11</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>13</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>23</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>385720</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56747987</td>\n",
       "      <td>56748187</td>\n",
       "      <td>42</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>385721</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56754063</td>\n",
       "      <td>56754271</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>15</td>\n",
       "      <td>5</td>\n",
       "      <td>12</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>9</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>385722</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56763412</td>\n",
       "      <td>56763626</td>\n",
       "      <td>231</td>\n",
       "      <td>6</td>\n",
       "      <td>116</td>\n",
       "      <td>196</td>\n",
       "      <td>376</td>\n",
       "      <td>187</td>\n",
       "      <td>94</td>\n",
       "      <td>...</td>\n",
       "      <td>97</td>\n",
       "      <td>45</td>\n",
       "      <td>4</td>\n",
       "      <td>58</td>\n",
       "      <td>191</td>\n",
       "      <td>51</td>\n",
       "      <td>0</td>\n",
       "      <td>24</td>\n",
       "      <td>128</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>385723</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56763699</td>\n",
       "      <td>56763918</td>\n",
       "      <td>27</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>9</td>\n",
       "      <td>16</td>\n",
       "      <td>13</td>\n",
       "      <td>7</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>385724</td>\n",
       "      <td>chrY</td>\n",
       "      <td>56847904</td>\n",
       "      <td>56848104</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>3</td>\n",
       "      <td>11</td>\n",
       "      <td>5</td>\n",
       "      <td>16</td>\n",
       "      <td>7</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>8</td>\n",
       "      <td>15</td>\n",
       "      <td>1</td>\n",
       "      <td>4</td>\n",
       "      <td>13</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>385725 rows × 415 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chrom     start       end  ADAD_CAUD_00_0281  ADAD_CAUD_00_0387  \\\n",
       "0       chr1     10015     10231                 12                 16   \n",
       "1       chr1    181363    181563                  1                  6   \n",
       "2       chr1    183716    183916                  7                  4   \n",
       "3       chr1    184083    184283                 11                  8   \n",
       "4       chr1    184370    184570                  7                  6   \n",
       "...      ...       ...       ...                ...                ...   \n",
       "385720  chrY  56747987  56748187                 42                  1   \n",
       "385721  chrY  56754063  56754271                 10                  0   \n",
       "385722  chrY  56763412  56763626                231                  6   \n",
       "385723  chrY  56763699  56763918                 27                  0   \n",
       "385724  chrY  56847904  56848104                  3                  7   \n",
       "\n",
       "        ADAD_CAUD_01_0164  ADAD_CAUD_01_1400  ADAD_CAUD_06_0194  \\\n",
       "0                      22                 12                 14   \n",
       "1                       4                  2                 10   \n",
       "2                       4                 12                  2   \n",
       "3                       4                 11                  2   \n",
       "4                       6                  0                  6   \n",
       "...                   ...                ...                ...   \n",
       "385720                  3                  3                  8   \n",
       "385721                 10                  9                 15   \n",
       "385722                116                196                376   \n",
       "385723                 10                  9                 16   \n",
       "385724                  3                 11                  5   \n",
       "\n",
       "        ADAD_CAUD_06_1486  ADAD_CAUD_07_0787  ...  LRRK_MDTG_01_39  \\\n",
       "0                      20                 12  ...                3   \n",
       "1                       6                  1  ...                1   \n",
       "2                      10                 19  ...                1   \n",
       "3                      19                 20  ...                1   \n",
       "4                       9                  6  ...                5   \n",
       "...                   ...                ...  ...              ...   \n",
       "385720                  5                  0  ...                0   \n",
       "385721                  5                 12  ...                8   \n",
       "385722                187                 94  ...               97   \n",
       "385723                 13                  7  ...                1   \n",
       "385724                 16                  7  ...                2   \n",
       "\n",
       "        LRRK_MDTG_04_10  LRRK_MDTG_10_37  LRRK_MDTG_13_60  LRRK_PTMN_01_39  \\\n",
       "0                     2               14               18                0   \n",
       "1                     5                2               14                0   \n",
       "2                    19               10               21                3   \n",
       "3                    20               26               25                3   \n",
       "4                    16               11               18                1   \n",
       "...                 ...              ...              ...              ...   \n",
       "385720                2                0                7                2   \n",
       "385721                2                0                5                9   \n",
       "385722               45                4               58              191   \n",
       "385723                4                0                4                2   \n",
       "385724                7                0                3                8   \n",
       "\n",
       "        LRRK_PTMN_04_10  LRRK_PTMN_10_37  LRRK_PTMN_13_60  LRRK_SUNI_04_10  \\\n",
       "0                     4               12                9               22   \n",
       "1                     4                7                4                8   \n",
       "2                    16                8                6               20   \n",
       "3                    22               18               11               26   \n",
       "4                    13                7                4               23   \n",
       "...                 ...              ...              ...              ...   \n",
       "385720                2                0                6                7   \n",
       "385721                5                0                1                4   \n",
       "385722               51                0               24              128   \n",
       "385723                9                0                1                1   \n",
       "385724               15                1                4               13   \n",
       "\n",
       "        LRRK_SUNI_10_37  \n",
       "0                    26  \n",
       "1                    18  \n",
       "2                    13  \n",
       "3                    24  \n",
       "4                    15  \n",
       "...                 ...  \n",
       "385720                0  \n",
       "385721                0  \n",
       "385722                6  \n",
       "385723                0  \n",
       "385724                5  \n",
       "\n",
       "[385725 rows x 415 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "idr_counts = pd.read_csv('/mnt/lab_data2/annashch/alzheimers_parkinsons/adpd.atac.idr.counts.txt.gz', sep='\\t')\n",
    "counts_header = idr_counts.columns\n",
    "display(idr_counts)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Intersect Counts Matrix with SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<BedTool(/tmp/pybedtools.yk6w1b6i.tmp)>"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_bed = pybedtools.BedTool.from_dataframe(idr_counts)\n",
    "counts_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<BedTool(/tmp/pybedtools.lin6tc81.tmp)>"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snps = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/snps/hg38_snps_1KG_ADPD.bed', header=None, sep='\\t')\n",
    "snps_bed = pybedtools.BedTool.from_dataframe(snps)\n",
    "snps_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "intersect_bed = counts_bed.intersect(snps_bed, u=True, wa=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    chrom      start        end  ADAD_CAUD_00_0281  ADAD_CAUD_00_0387  \\\n",
      "0    chr1   39878006   39878581                 51                 44   \n",
      "1    chr1   39878599   39879244                 57                 69   \n",
      "2    chr1   39882784   39884105                302                302   \n",
      "3    chr1   39887608   39887837                 11                 12   \n",
      "4    chr1  154865411  154865652                 45                 19   \n",
      "..    ...        ...        ...                ...                ...   \n",
      "607  chr9   17699014   17699637                 52                 27   \n",
      "608  chr9   17730813   17731948                 34                 57   \n",
      "609  chr9   33816144   33818416               1039               1523   \n",
      "610  chr9   33957284   33957573                 18                  6   \n",
      "611  chr9   33971219   33972644                 86                 23   \n",
      "\n",
      "     ADAD_CAUD_01_0164  ADAD_CAUD_01_1400  ADAD_CAUD_06_0194  \\\n",
      "0                   62                 94                 37   \n",
      "1                  114                107                 42   \n",
      "2                  616                325                169   \n",
      "3                    9                  6                  8   \n",
      "4                   21                 25                 18   \n",
      "..                 ...                ...                ...   \n",
      "607                 15                 26                 40   \n",
      "608                 12                 57                 66   \n",
      "609               3440               1911                635   \n",
      "610                 11                 14                  9   \n",
      "611                 19                 22                 56   \n",
      "\n",
      "     ADAD_CAUD_06_1486  ADAD_CAUD_07_0787  ...  LRRK_MDTG_01_39  \\\n",
      "0                   97                 62  ...               14   \n",
      "1                  125                102  ...               16   \n",
      "2                  438                484  ...               65   \n",
      "3                    6                 15  ...               17   \n",
      "4                   10                 52  ...               14   \n",
      "..                 ...                ...  ...              ...   \n",
      "607                 14                 53  ...               46   \n",
      "608                 59                 54  ...              106   \n",
      "609               1769               1845  ...              261   \n",
      "610                  6                 14  ...                5   \n",
      "611                 26                 67  ...               52   \n",
      "\n",
      "     LRRK_MDTG_04_10  LRRK_MDTG_10_37  LRRK_MDTG_13_60  LRRK_PTMN_01_39  \\\n",
      "0                 70              111               40               14   \n",
      "1                 81              214               43               30   \n",
      "2                305              414              307               75   \n",
      "3                 31               34               60               12   \n",
      "4                 40               33               87               24   \n",
      "..               ...              ...              ...              ...   \n",
      "607               69               61               77               37   \n",
      "608              224              138              284               32   \n",
      "609             1102             1466             1225              224   \n",
      "610               17                9               13               24   \n",
      "611               95              113              143               71   \n",
      "\n",
      "     LRRK_PTMN_04_10  LRRK_PTMN_10_37  LRRK_PTMN_13_60  LRRK_SUNI_04_10  \\\n",
      "0                170               54               91               56   \n",
      "1                196               80              131              108   \n",
      "2                534              337              302              413   \n",
      "3                 58               50               72                3   \n",
      "4                117              125               93               20   \n",
      "..               ...              ...              ...              ...   \n",
      "607               88               78               97               46   \n",
      "608               84               55               91               59   \n",
      "609             1243             1070              747             1650   \n",
      "610               62               40               56                9   \n",
      "611              229              190              319               26   \n",
      "\n",
      "     LRRK_SUNI_10_37  \n",
      "0                 87  \n",
      "1                110  \n",
      "2                332  \n",
      "3                 17  \n",
      "4                 33  \n",
      "..               ...  \n",
      "607               40  \n",
      "608               59  \n",
      "609             1338  \n",
      "610               15  \n",
      "611               57  \n",
      "\n",
      "[612 rows x 415 columns]\n"
     ]
    }
   ],
   "source": [
    "idr_counts = pybedtools.BedTool.to_dataframe(intersect_bed, header=None)\n",
    "idr_counts.columns = counts_header \n",
    "print(idr_counts)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create Region-Specific Counts Matrices"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAUD\n",
      "['CTRH_CAUD_07_1144', 'CTRH_CAUD_07_1287', 'CTRH_CAUD_08_0712', 'CTRH_CAUD_10_0867', 'CTRH_CAUD_13_0962', 'CTRH_CAUD_14_0380', 'CTRH_CAUD_14_0513', 'CTRH_CAUD_14_1383', 'CTRH_CAUD_15_1023', 'CTRH_CAUD_15_1025', 'CTRL_CAUD_06_0615', 'CTRL_CAUD_06_1516', 'CTRL_CAUD_09_1589', 'CTRL_CAUD_11_0393', 'CTRL_CAUD_13_0038', 'CTRL_CAUD_13_1226', 'CTRL_CAUD_14_0586', 'CTRL_CAUD_14_0941', 'CTRL_CAUD_14_1018', 'CTRL_CAUD_00_38', 'CTRL_CAUD_01_31', 'CTRL_CAUD_03_15', 'CTRL_CAUD_03_39', 'CTRL_CAUD_03_41', 'CTRL_CAUD_03_66', 'CTRL_CAUD_04_38', 'CTRL_CAUD_05_16', 'CTRL_CAUD_08_90', 'CTRL_CAUD_09_35', 'CTRL_CAUD_15_78', 'CTRL_CAUD_16_10', 'CTRL_CAUD_16_32']\n",
      "HIPP\n",
      "['CTRH_HIPP_07_1058', 'CTRH_HIPP_07_1144', 'CTRH_HIPP_07_1287', 'CTRH_HIPP_08_0298', 'CTRH_HIPP_10_0867', 'CTRH_HIPP_13_0962', 'CTRH_HIPP_14_0380', 'CTRH_HIPP_14_0513', 'CTRH_HIPP_14_1383', 'CTRH_HIPP_15_1023', 'CTRH_HIPP_15_1025', 'CTRL_HIPP_06_1516', 'CTRL_HIPP_11_0393', 'CTRL_HIPP_13_0038', 'CTRL_HIPP_13_1226', 'CTRL_HIPP_14_0586', 'CTRL_HIPP_14_0941', 'CTRL_HIPP_14_1018', 'CTRL_HIPP_00_38', 'CTRL_HIPP_04_38', 'CTRL_HIPP_05_16', 'CTRL_HIPP_09_35']\n",
      "MDFG\n",
      "['CTRL_MDFG_00_38', 'CTRL_MDFG_01_37', 'CTRL_MDFG_03_39', 'CTRL_MDFG_03_41', 'CTRL_MDFG_03_66', 'CTRL_MDFG_04_38', 'CTRL_MDFG_05_16', 'CTRL_MDFG_09_35', 'CTRL_MDFG_09_50', 'CTRL_MDFG_15_78', 'CTRL_MDFG_16_10', 'CTRL_MDFG_16_32']\n",
      "MDTG\n",
      "['CTRL_MDTG_00_38', 'CTRL_MDTG_03_15', 'CTRL_MDTG_03_39', 'CTRL_MDTG_03_41', 'CTRL_MDTG_03_66', 'CTRL_MDTG_04_38', 'CTRL_MDTG_05_16', 'CTRL_MDTG_09_35', 'CTRL_MDTG_09_50', 'CTRL_MDTG_15_78', 'CTRL_MDTG_16_10', 'CTRL_MDTG_16_32']\n",
      "PTMN\n",
      "['CTRL_PTMN_00_38', 'CTRL_PTMN_03_39', 'CTRL_PTMN_03_41', 'CTRL_PTMN_03_66', 'CTRL_PTMN_04_38', 'CTRL_PTMN_05_16', 'CTRL_PTMN_07_28', 'CTRL_PTMN_09_35', 'CTRL_PTMN_09_50', 'CTRL_PTMN_15_78', 'CTRL_PTMN_16_32']\n",
      "SUNI\n",
      "['CTRL_SUNI_00_38', 'CTRL_SUNI_01_31', 'CTRL_SUNI_03_15', 'CTRL_SUNI_03_39', 'CTRL_SUNI_03_41', 'CTRL_SUNI_03_66', 'CTRL_SUNI_04_38', 'CTRL_SUNI_05_16', 'CTRL_SUNI_06_57', 'CTRL_SUNI_07_28', 'CTRL_SUNI_09_35', 'CTRL_SUNI_09_50', 'CTRL_SUNI_15_78', 'CTRL_SUNI_16_32']\n",
      "SMTG\n",
      "['CTRH_SMTG_07_1058', 'CTRH_SMTG_07_1144', 'CTRH_SMTG_07_1287', 'CTRH_SMTG_08_0712', 'CTRH_SMTG_10_0867', 'CTRH_SMTG_13_0962', 'CTRH_SMTG_14_0380', 'CTRH_SMTG_14_0513', 'CTRH_SMTG_14_1383', 'CTRH_SMTG_15_1023', 'CTRH_SMTG_15_1025', 'CTRL_SMTG_06_1516', 'CTRL_SMTG_09_1589', 'CTRL_SMTG_11_0311', 'CTRL_SMTG_11_0393', 'CTRL_SMTG_13_0038', 'CTRL_SMTG_13_1226', 'CTRL_SMTG_14_0586', 'CTRL_SMTG_14_1018']\n",
      "PARL\n",
      "['CTRH_PARL_07_1058', 'CTRH_PARL_07_1144', 'CTRH_PARL_07_1287', 'CTRH_PARL_08_0298', 'CTRH_PARL_10_0867', 'CTRH_PARL_13_0962', 'CTRH_PARL_14_0380', 'CTRH_PARL_14_0513', 'CTRH_PARL_14_1383', 'CTRH_PARL_15_1023', 'CTRH_PARL_15_1025', 'CTRL_PARL_09_1589', 'CTRL_PARL_11_0311', 'CTRL_PARL_11_0393', 'CTRL_PARL_13_0038', 'CTRL_PARL_13_1226', 'CTRL_PARL_14_0586', 'CTRL_PARL_14_1018']\n"
     ]
    }
   ],
   "source": [
    "for reg in multi_bam_regs:\n",
    "    print(reg)\n",
    "    with open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/bam_lists/' + reg + '_ctrl_bams.txt') as infile:\n",
    "        reg_bams = [i.strip() for i in infile.readlines()]\n",
    "    bam_names = [bam.split('/')[10] for bam in reg_bams]\n",
    "    print(bam_names)\n",
    "    reg_counts = idr_counts.copy(deep=True)\n",
    "    reg_counts['index'] = reg_counts['chrom'] + '_' + reg_counts['start'].astype(str) + '_' + reg_counts['end'].astype(str)\n",
    "    reg_counts = reg_counts[['index'] + bam_names]\n",
    "    reg_counts.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                        + reg + '/idr_counts.txt', index=False, header=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Create Chromosome-Specific Inputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['chr1', 'chr2', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', 'chr19', 'chr20', 'chr21', 'chr22', 'chrX', 'chrY']\n",
      "CAUD\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "HIPP\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "MDFG\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "MDTG\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "PTMN\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "SUNI\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "SMTG\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "PARL\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n"
     ]
    }
   ],
   "source": [
    "chroms = ['chr' + str(i) for i in range(1, 23)]\n",
    "chroms.append('chrX')\n",
    "chroms.append('chrY')\n",
    "print(chroms)\n",
    "for reg in multi_bam_regs:\n",
    "    print(reg)\n",
    "    reg_counts = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                        + reg + '/idr_counts.txt', header=None, sep='\\t')\n",
    "    reg_offset = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                        + reg + '/offset.txt',header=None, sep='\\t')\n",
    "    for chrom in chroms:\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                             + reg + '/' + chrom):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                     + reg + '/' + chrom)\n",
    "        print(chrom)\n",
    "        reg_chrom_counts = reg_counts.loc[reg_counts[0].apply(lambda x : x.startswith(chrom))]\n",
    "        reg_chrom_offset = reg_offset.loc[reg_offset[0].apply(lambda x : x.startswith(chrom))]\n",
    "        reg_chrom_counts.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                                 + reg + '/' + chrom + '/Y.txt', index=False, header=False, sep='\\t')\n",
    "        reg_chrom_offset.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                                 + reg + '/' + chrom + '/K.txt', index=False, header=False, sep='\\t')\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get Genotyped ASVCFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAUD\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "HIPP\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "MDFG\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "MDTG\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "PTMN\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "SUNI\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "SMTG\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "PARL\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n"
     ]
    }
   ],
   "source": [
    "for reg in multi_bam_regs:\n",
    "    print(reg)\n",
    "    with open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/bam_lists/' + reg + '_ctrl_bams.txt') as infile:\n",
    "        reg_bams = [i.strip() for i in infile.readlines()]\n",
    "    asvcf = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/asvcf.gz',\n",
    "                        sep='\\t', skiprows=1, header=None)\n",
    "    bam_names = [bam.split('/')[10] for bam in reg_bams]\n",
    "    bam_patients = [name.split('_')[2] + '_' + name.split('_')[3] for name in bam_names]\n",
    "    header = ['chrom', 'pos', 'rsid', 'ref', 'alt', 'score', 'qc', 'info', 'format'] + bam_patients\n",
    "    asvcf.columns = header\n",
    "    asvcf.set_index('rsid', inplace=True)\n",
    "    #print(asvcf.head())\n",
    "    for patient in bam_patients:\n",
    "        if os.path.isfile('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/quasar/output/'\n",
    "                            + patient + '/genotypes.txt'):\n",
    "            genotype = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/quasar/output/'\n",
    "                                   + patient + '/genotypes.txt', sep='\\t')\n",
    "        genotype = genotype[['rsID', 'map.g0', 'map.g1', 'map.g2']]\n",
    "        for index,row in genotype.iterrows():\n",
    "            if row['map.g1'] >= row['map.g0']:\n",
    "                if row['map.g1'] >= row['map.g2']:\n",
    "                    asvcf.at[row['rsID'], patient] = asvcf.at[row['rsID'], patient].replace('./.', '0/1')\n",
    "                else:\n",
    "                    asvcf.at[row['rsID'], patient] = asvcf.at[row['rsID'], patient].replace('./.', '1/1')\n",
    "            else:\n",
    "                if row['map.g0'] >= row['map.g2']:\n",
    "                    asvcf.at[row['rsID'], patient] = asvcf.at[row['rsID'], patient].replace('./.', '0/0')\n",
    "                else:\n",
    "                    asvcf.at[row['rsID'], patient] = asvcf.at[row['rsID'], patient].replace('./.', '1/1')\n",
    "    asvcf.reset_index(inplace=True)\n",
    "    asvcf = asvcf[header]\n",
    "    #print(asvcf.head())\n",
    "    asvcf.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/genotyped_asvcf.txt',\n",
    "                    sep='\\t', index=False)\n",
    "    for chrom in chroms:\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                             + reg + '/' + chrom):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                     + reg + '/' + chrom)\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/'\n",
    "                             + reg + '/' + chrom):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/output/'\n",
    "                     + reg + '/' + chrom)\n",
    "        print(chrom)\n",
    "        chrom_asvcf = asvcf.loc[asvcf['chrom'] == chrom]\n",
    "        chrom_asvcf.to_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                                 + reg + '/' + chrom + '/asvcf', index=False, header=False, sep='\\t')\n",
    "        bgzip_cmd = 'bgzip /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/' + chrom + '/asvcf'\n",
    "        ! {bgzip_cmd}\n",
    "        tabix_cmd = 'tabix -p vcf /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' + reg + '/' + chrom + '/asvcf.gz'\n",
    "        ! {tabix_cmd}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAUD\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "HIPP\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "MDFG\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "MDTG\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "PTMN\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "SUNI\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "SMTG\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "PARL\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n"
     ]
    }
   ],
   "source": [
    "for reg in multi_bam_regs:\n",
    "    print(reg)\n",
    "    for chrom in chroms:\n",
    "        print(chrom)\n",
    "        if not os.path.isdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/'\n",
    "                             + reg + '/' + chrom):\n",
    "            os.mkdir('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/'\n",
    "                     + reg + '/' + chrom)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate RASQUAL commands"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAUD\n",
      "chr1\n",
      "     0         1           2  3  4    5     6   \\\n",
      "0  chr1  39822151  rs34640847  G  A  100  PASS   \n",
      "1  chr1  39822245  rs36015266  C  T  100  PASS   \n",
      "2  chr1  39822260  rs61779808  T  A  100  PASS   \n",
      "3  chr1  39822792  rs61779809  C  T  100  PASS   \n",
      "4  chr1  39825225  rs72666941  C  A  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=595;AF=0.11881;AN=5008;NS=2504;DP=18110;EAS...  GT:AS  ./.:0,0  ...   \n",
      "1  AC=595;AF=0.11881;AN=5008;NS=2504;DP=19296;EAS...  GT:AS  ./.:0,0  ...   \n",
      "2  AC=493;AF=0.0984425;AN=5008;NS=2504;DP=19241;E...  GT:AS  ./.:1,0  ...   \n",
      "3  AC=595;AF=0.11881;AN=5008;NS=2504;DP=19613;EAS...  GT:AS  ./.:1,0  ...   \n",
      "4  AC=477;AF=0.0952476;AN=5008;NS=2504;DP=20561;E...  GT:AS  ./.:1,0  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0   \n",
      "1  ./.:0,1  ./.:1,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:0,0   \n",
      "2  ./.:0,1  ./.:1,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,1  ./.:0,0   \n",
      "3  ./.:0,0  ./.:1,0  ./.:0,0  ./.:2,0  ./.:1,0  ./.:0,0  ./.:1,0  ./.:1,0   \n",
      "4  ./.:2,0  ./.:2,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:0,0  ./.:1,0  ./.:1,0   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:0,0  ./.:0,0  \n",
      "1  ./.:1,0  ./.:0,1  \n",
      "2  ./.:1,0  ./.:0,1  \n",
      "3  ./.:0,0  ./.:0,0  \n",
      "4  ./.:1,0  ./.:1,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                         0    1    2    3    4    5    6    7    8    9   ...  \\\n",
      "0    chr1_39878006_39878581   54   37   98   68   69  123   55   69   52  ...   \n",
      "1    chr1_39878599_39879244   81   54  141   90  130  191   59   95   68  ...   \n",
      "2    chr1_39882784_39884105  297  169  566  701  525  546  339  472  548  ...   \n",
      "3    chr1_39887608_39887837   61   34   44   88   96   73   64   98   77  ...   \n",
      "4  chr1_154865411_154865652  122   77   72  259  244  167  176  226  332  ...   \n",
      "\n",
      "    23   24   25   26   27   28   29   30   31   32  \n",
      "0   92   51   87   84   60   18  133   73  111   64  \n",
      "1  125   58  128  122   80   29  169   76  186   73  \n",
      "2  446  349  394  578  450  106  739  454  683  283  \n",
      "3  105   59   34   93   77    7  100   59   73   67  \n",
      "4  228  134  137  210  239   41  261  193  188  188  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  246\n",
      "Feature SNPs (m):  380\n",
      "chr2\n",
      "     0         1           2  3  4    5     6   \\\n",
      "0  chr2  17966582  rs76116224  A  T  100  PASS   \n",
      "1  chr2  95050122   rs3112982  T  C  100  PASS   \n",
      "2  chr2  95050899   rs3112983  G  C  100  PASS   \n",
      "3  chr2  95054885   rs2291991  C  A  100  PASS   \n",
      "4  chr2  95057164    rs872580  C  A  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=199;AF=0.0397364;AN=5008;NS=2504;DP=19732;E...  GT:AS  0/0:0,0  ...   \n",
      "1  AC=3988;AF=0.796326;AN=5008;NS=2504;DP=19103;E...  GT:AS  0/0:0,0  ...   \n",
      "2  AC=3867;AF=0.772165;AN=5008;NS=2504;DP=16896;E...  GT:AS  ./.:0,0  ...   \n",
      "3  AC=3935;AF=0.785743;AN=5008;NS=2504;DP=18455;E...  GT:AS  0/0:2,0  ...   \n",
      "4  AC=3913;AF=0.78135;AN=5008;NS=2504;DP=19307;EA...  GT:AS  0/0:1,0  ...   \n",
      "\n",
      "        31        32        33        34       35       36        37  \\\n",
      "0  ./.:0,0   0/0:0,0   0/0:0,0   0/0:0,0  ./.:1,0  ./.:2,0   0/0:3,0   \n",
      "1  0/1:0,0   ./.:0,1   1/1:0,2   ./.:0,2  ./.:1,0  ./.:0,1   ./.:0,3   \n",
      "2  0/1:3,0   ./.:2,1   ./.:0,0   1/1:0,1  ./.:0,0  ./.:0,3   1/1:0,6   \n",
      "3  0/1:1,1   0/1:1,1   ./.:0,5   ./.:0,2  0/1:2,3  0/1:0,7   1/1:0,5   \n",
      "4  0/1:9,6  0/1:6,13  1/1:0,17  1/1:0,17  0/1:9,4  0/1:0,8  1/1:0,28   \n",
      "\n",
      "         38       39       40  \n",
      "0   ./.:0,0  ./.:1,0  ./.:0,1  \n",
      "1   ./.:0,0  ./.:1,1  ./.:0,0  \n",
      "2   0/0:2,0  ./.:0,1  ./.:1,0  \n",
      "3   ./.:5,0  ./.:1,2  0/0:2,0  \n",
      "4  0/0:20,0  0/1:6,9  0/0:7,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                       0    1    2     3     4     5     6    7    8    9   \\\n",
      "0  chr2_17966403_17966603    9    2     8     3    12     8    5    9    5   \n",
      "1  chr2_95056576_95058298  332  154  1609  1378   699  1564  550  527  571   \n",
      "2  chr2_95063135_95063337   12    8    67    59    32    84   23   40   18   \n",
      "3  chr2_95071721_95071943   26    9    81    38    33    79   28   17   22   \n",
      "4  chr2_95076777_95078568  589  289   658  1042  1049  1171  705  817  943   \n",
      "\n",
      "   ...   23   24    25    26   27   28    29    30    31   32  \n",
      "0  ...    9   10     8    11   11    9    14    10     7   12  \n",
      "1  ...  731  540  1203  1307  832  226  1689  1036  1480  517  \n",
      "2  ...   38   24    51    48   33   22    74    51    59   24  \n",
      "3  ...   29   29    23    40   46    3    60    36    60   30  \n",
      "4  ...  826  501   793   911  741  210  1024   943  1333  532  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  409\n",
      "Feature SNPs (m):  62\n",
      "chr3\n",
      "     0         1           2  3  4    5     6   \\\n",
      "0  chr3  18163160  rs35228011  G  A  100  PASS   \n",
      "1  chr3  18163667  rs35115317  G  A  100  PASS   \n",
      "2  chr3  18168316  rs67054573  T  A  100  PASS   \n",
      "3  chr3  18169834  rs13080765  G  A  100  PASS   \n",
      "4  chr3  18171934  rs13092771  C  T  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=312;AF=0.0623003;AN=5008;NS=2504;DP=18623;E...  GT:AS  0/0:0,0  ...   \n",
      "1  AC=312;AF=0.0623003;AN=5008;NS=2504;DP=16716;E...  GT:AS  0/0:0,0  ...   \n",
      "2  AC=321;AF=0.0640974;AN=5008;NS=2504;DP=20057;E...  GT:AS  ./.:0,0  ...   \n",
      "3  AC=312;AF=0.0623003;AN=5008;NS=2504;DP=15733;E...  GT:AS  ./.:0,0  ...   \n",
      "4  AC=320;AF=0.0638978;AN=5008;NS=2504;DP=15767;E...  GT:AS  ./.:0,0  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:0,0  ./.:0,0   \n",
      "1  ./.:2,1  ./.:1,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:1,1  ./.:0,2  0/0:0,0   \n",
      "2  ./.:0,0  ./.:2,0  ./.:0,0  ./.:1,0  ./.:3,0  ./.:0,0  ./.:0,1  ./.:0,0   \n",
      "3  ./.:0,0  ./.:1,0  ./.:0,2  ./.:2,0  ./.:1,0  ./.:2,1  ./.:0,0  ./.:2,0   \n",
      "4  ./.:1,0  ./.:0,0  ./.:1,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:0,1  ./.:2,0   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:1,0  ./.:1,0  \n",
      "1  ./.:1,0  ./.:3,0  \n",
      "2  ./.:1,0  ./.:1,0  \n",
      "3  ./.:1,0  ./.:1,0  \n",
      "4  ./.:1,0  ./.:0,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                       0    1    2    3    4    5    6    7    8     9   ...  \\\n",
      "0  chr3_18177287_18177743   76   26   26  107   91   51   41   87   125  ...   \n",
      "1  chr3_18189793_18189997   34   14   21   40   43   29   23   40    31  ...   \n",
      "2  chr3_18222391_18223069   40   22   50   29   36   43   25   19    47  ...   \n",
      "3  chr3_48685519_48686987  539  271  713  966  835  647  583  613  1218  ...   \n",
      "4  chr3_48692599_48693032  121   56  162  185  236  150  106  143   219  ...   \n",
      "\n",
      "    23   24   25   26   27   28   29   30    31   32  \n",
      "0   85   46   50  139  136   40  105   55    55   77  \n",
      "1   49   22   33   30   56   15   83   37    23   43  \n",
      "2   46   27   31   40  103   28   65   52    64   43  \n",
      "3  739  374  563  599  522  151  719  772  1146  306  \n",
      "4  186  118  141  191  121   55  228  173   246  141  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  635\n",
      "Feature SNPs (m):  47\n",
      "chr4\n",
      "     0       1           2  3  4    5     6   \\\n",
      "0  chr4  849886  rs55728587  G  A  100  PASS   \n",
      "1  chr4  853657  rs34006051  C  T  100  PASS   \n",
      "2  chr4  857569  rs11731387  G  A  100  PASS   \n",
      "3  chr4  858525   rs1564282  C  T  100  PASS   \n",
      "4  chr4  862999  rs12651368  G  A  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=466;AF=0.0930511;AN=5008;NS=2504;DP=10947;E...  GT:AS  ./.:0,0  ...   \n",
      "1  AC=478;AF=0.0954473;AN=5008;NS=2504;DP=12128;E...  GT:AS  ./.:3,0  ...   \n",
      "2  AC=535;AF=0.106829;AN=5008;NS=2504;DP=14139;EA...  GT:AS  ./.:1,0  ...   \n",
      "3  AC=472;AF=0.0942492;AN=5008;NS=2504;DP=17354;E...  GT:AS  0/0:1,0  ...   \n",
      "4  AC=483;AF=0.0964457;AN=5008;NS=2504;DP=19702;E...  GT:AS  ./.:3,0  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:0,0  ./.:1,0  ./.:0,0  ./.:0,1  ./.:0,0  ./.:1,0  ./.:0,0  ./.:2,0   \n",
      "1  ./.:0,0  ./.:1,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:2,0  ./.:1,0   \n",
      "2  ./.:0,0  ./.:2,0  ./.:0,0  0/1:3,2  ./.:0,0  ./.:1,0  ./.:3,0  ./.:3,0   \n",
      "3  ./.:1,0  ./.:1,0  ./.:2,0  ./.:1,1  0/0:2,0  0/0:2,0  0/0:5,0  ./.:1,0   \n",
      "4  ./.:2,0  ./.:5,0  ./.:0,0  ./.:1,1  ./.:1,0  ./.:2,0  0/0:6,0  ./.:1,0   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:0,0  ./.:0,0  \n",
      "1  ./.:0,1  ./.:0,0  \n",
      "2  ./.:1,0  ./.:2,0  \n",
      "3  ./.:1,0  ./.:0,0  \n",
      "4  ./.:0,1  ./.:1,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                   0    1    2    3    4    5    6    7    8     9   ...   23  \\\n",
      "0  chr4_875302_876377  142   95   78  217  220  131  132  222   249  ...  225   \n",
      "1  chr4_885987_886717   13   17   23   38   44   25   47   31    33  ...   49   \n",
      "2  chr4_898875_900016  125   64   73  181  191  155  100  167   217  ...  197   \n",
      "3  chr4_931456_933167  454  210  905  945  675  750  549  603  1096  ...  577   \n",
      "4  chr4_957451_958607  133   54  197  202  204  161  161  187   307  ...  201   \n",
      "\n",
      "    24   25   26   27   28   29   30    31   32  \n",
      "0  143  120  146  193   49  253  165   211  101  \n",
      "1   28   31   55   34   13   48   28    26   23  \n",
      "2  123   94  112  144   29  151  132   270  113  \n",
      "3  364  595  563  552  160  551  639  1031  255  \n",
      "4  121  137  187  133   51  202  216   267   99  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  739\n",
      "Feature SNPs (m):  44\n",
      "chr5\n",
      "     0         1          2  3  4    5     6   \\\n",
      "0  chr5  60753292  rs2219333  T  C  100  PASS   \n",
      "1  chr5  60756742  rs1031744  C  A  100  PASS   \n",
      "2  chr5  60762046  rs6894261  T  C  100  PASS   \n",
      "3  chr5  60762105  rs1563522  A  G  100  PASS   \n",
      "4  chr5  60762571  rs1031745  C  T  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=923;AF=0.184305;AN=5008;NS=2504;DP=12265;EA...  GT:AS  ./.:1,0  ...   \n",
      "1  AC=882;AF=0.176118;AN=5008;NS=2504;DP=15760;EA...  GT:AS  ./.:0,0  ...   \n",
      "2  AC=921;AF=0.183906;AN=5008;NS=2504;DP=16634;EA...  GT:AS  ./.:0,0  ...   \n",
      "3  AC=987;AF=0.197085;AN=5008;NS=2504;DP=16311;EA...  GT:AS  ./.:2,0  ...   \n",
      "4  AC=986;AF=0.196885;AN=5008;NS=2504;DP=15665;EA...  GT:AS  ./.:1,0  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:2,0  ./.:0,0  ./.:0,0   \n",
      "1  ./.:0,0  ./.:1,0  ./.:2,0  ./.:0,0  ./.:1,0  ./.:1,0  ./.:1,0  ./.:1,0   \n",
      "2  ./.:2,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:2,0  ./.:2,0  ./.:2,0   \n",
      "3  ./.:1,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:2,0  ./.:1,0  ./.:1,0  ./.:2,0   \n",
      "4  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:2,0  ./.:0,0  ./.:0,0   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:0,0  ./.:0,0  \n",
      "1  ./.:0,0  ./.:1,0  \n",
      "2  ./.:0,0  ./.:2,0  \n",
      "3  ./.:0,2  ./.:0,0  \n",
      "4  ./.:0,0  ./.:0,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                         0    1    2    3     4     5     6    7    8     9   \\\n",
      "0    chr5_60842597_60842797   44   18   23    28    34    56   27   35    24   \n",
      "1    chr5_60902867_60904436  116   45  220   148   175   330  108   94   161   \n",
      "2    chr5_60944432_60946454  764  390  915  1198  1172  1044  776  997  1131   \n",
      "3  chr5_102909954_102911147  184  129  121   202   270   180  131  237   288   \n",
      "4  chr5_102959118_102959383   61   43   66   156   142   105   76  130   192   \n",
      "\n",
      "   ...    23   24   25    26    27   28    29   30    31   32  \n",
      "0  ...    48   19   35    34    62   24    60   50    60   48  \n",
      "1  ...   187   85  224   178   351  102   329  212   178  152  \n",
      "2  ...  1009  724  868  1064  1508  382  1463  958  1550  674  \n",
      "3  ...   295  181  121   218   420   96   283  200   246  275  \n",
      "4  ...   123   80  100   156   224   49   173  123    74  118  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  425\n",
      "Feature SNPs (m):  24\n",
      "chr6\n",
      "     0         1           2  3  4    5     6   \\\n",
      "0  chr6  27707511   rs1139226  A  T  100  PASS   \n",
      "1  chr6  27708530   rs9468195  A  G  100  PASS   \n",
      "2  chr6  27709015   rs4713118  A  G  100  PASS   \n",
      "3  chr6  27709949  rs13215072  A  G  100  PASS   \n",
      "4  chr6  27713436   rs9468199  G  A  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=1490;AF=0.297524;AN=5008;NS=2504;DP=16934;E...  GT:AS  ./.:0,0  ...   \n",
      "1  AC=1491;AF=0.297724;AN=5008;NS=2504;DP=16808;E...  GT:AS  ./.:1,0  ...   \n",
      "2  AC=1556;AF=0.310703;AN=5008;NS=2504;DP=16928;E...  GT:AS  ./.:1,0  ...   \n",
      "3  AC=1556;AF=0.310703;AN=5008;NS=2504;DP=14198;E...  GT:AS  ./.:0,0  ...   \n",
      "4  AC=1504;AF=0.300319;AN=5008;NS=2504;DP=18162;E...  GT:AS  ./.:0,0  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,1  ./.:0,0  ./.:0,0   \n",
      "1  ./.:0,0  ./.:3,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:0,0   \n",
      "2  ./.:0,0  ./.:2,0  ./.:0,0  ./.:2,0  ./.:0,0  ./.:0,0  ./.:3,0  ./.:0,0   \n",
      "3  ./.:0,0  ./.:1,0  ./.:1,0  ./.:1,0  ./.:0,1  ./.:0,2  ./.:0,0  ./.:2,0   \n",
      "4  ./.:0,0  ./.:2,0  ./.:2,0  ./.:1,0  ./.:1,0  ./.:0,0  ./.:2,0  ./.:2,0   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:0,0  ./.:0,0  \n",
      "1  ./.:0,0  ./.:0,0  \n",
      "2  ./.:0,0  ./.:0,0  \n",
      "3  ./.:1,0  ./.:0,1  \n",
      "4  ./.:1,0  ./.:1,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                       0    1    2    3    4    5    6    7    8    9   ...  \\\n",
      "0  chr6_27727369_27727569   18   15    4   35    9    9   19   14   29  ...   \n",
      "1  chr6_27751882_27753033  303  156  316  542  471  481  288  410  518  ...   \n",
      "2  chr6_27754110_27754376  107   31   90  167  111  118  115  126  155  ...   \n",
      "3  chr6_27761922_27762971   58   34  139  230   63   68   79  115  157  ...   \n",
      "4  chr6_27764857_27765356   78   40   63  118  138   65   67  149  190  ...   \n",
      "\n",
      "    23   24   25   26   27  28   29   30   31   32  \n",
      "0   12    9    3   23   20   6   17    6    5   15  \n",
      "1  512  328  371  511  500  66  638  434  500  356  \n",
      "2  128   91   89  133  164  26  192  106  101   95  \n",
      "3  101   56   46  135  229  38  126   67  151   97  \n",
      "4  134   85   62  111  146  28  155  123  114  128  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  184\n",
      "Feature SNPs (m):  10\n",
      "chr7\n",
      "     0         1           2  3  4    5     6   \\\n",
      "0  chr7  23071112   rs7797870  C  G  100  PASS   \n",
      "1  chr7  23073884   rs6461685  C  T  100  PASS   \n",
      "2  chr7  23076240  rs12539467  A  G  100  PASS   \n",
      "3  chr7  23079437   rs7811903  A  G  100  PASS   \n",
      "4  chr7  23079688   rs7776644  C  T  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=1573;AF=0.314097;AN=5008;NS=2504;DP=16197;E...  GT:AS  ./.:1,0  ...   \n",
      "1  AC=1536;AF=0.306709;AN=5008;NS=2504;DP=14661;E...  GT:AS  0/0:1,0  ...   \n",
      "2  AC=1602;AF=0.319888;AN=5008;NS=2504;DP=19948;E...  GT:AS  ./.:2,0  ...   \n",
      "3  AC=2086;AF=0.416534;AN=5008;NS=2504;DP=14374;E...  GT:AS  ./.:1,0  ...   \n",
      "4  AC=2203;AF=0.439896;AN=5008;NS=2504;DP=16920;E...  GT:AS  ./.:1,0  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:0,2  ./.:0,2  ./.:0,0  0/0:0,0  ./.:2,0  ./.:1,0  ./.:0,0  ./.:0,0   \n",
      "1  ./.:2,1  0/1:4,1  0/0:2,0  0/0:4,0  ./.:2,0  ./.:1,0  ./.:3,0  0/1:0,1   \n",
      "2  ./.:0,0  ./.:0,0  ./.:1,0  ./.:0,0  ./.:2,0  ./.:3,2  ./.:0,0  ./.:1,1   \n",
      "3  ./.:0,0  ./.:3,0  ./.:1,0  ./.:1,0  ./.:0,0  ./.:0,0  ./.:3,0  ./.:0,0   \n",
      "4  ./.:1,0  ./.:0,1  ./.:1,0  ./.:2,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:0,0   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:0,0  ./.:0,0  \n",
      "1  ./.:3,0  ./.:1,0  \n",
      "2  ./.:0,0  ./.:1,0  \n",
      "3  ./.:0,0  ./.:0,0  \n",
      "4  ./.:0,0  ./.:0,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                       0    1    2     3     4     5     6    7     8     9   \\\n",
      "0  chr7_23085842_23086307   64   61    39    66   101    60   70    95   116   \n",
      "1  chr7_23105000_23105585  194   82   157   263   235   206  156   260   275   \n",
      "2  chr7_23105594_23106828  773  375  1014  1492  1333  1221  947  1125  1900   \n",
      "3  chr7_23114357_23115183  105   59    94   152   186   109  100   113   232   \n",
      "4  chr7_23115998_23116546   70   44    38    71    90    74   53    62    91   \n",
      "\n",
      "   ...    23   24    25    26    27   28    29    30    31   32  \n",
      "0  ...    84   58    32    74   119   45   104    62   102   93  \n",
      "1  ...   189  134   192   208   328   95   294   237   292  173  \n",
      "2  ...  1145  701  1087  1395  1783  459  1572  1230  1730  821  \n",
      "3  ...   158  103    88   149   201   67   248   123   205  217  \n",
      "4  ...    84   74    63    76   108   49   105    71   104  107  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  369\n",
      "Feature SNPs (m):  21\n",
      "chr8\n",
      "     0         1          2  3  4    5     6   \\\n",
      "0  chr8  11836068  rs1293325  G  A  100  PASS   \n",
      "1  chr8  11837477  rs1293327  G  A  100  PASS   \n",
      "2  chr8  11840927  rs1293331  C  T  100  PASS   \n",
      "3  chr8  11843744  rs2740592  A  G  100  PASS   \n",
      "4  chr8  11843769  rs2645425  G  A  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=623;AF=0.124401;AN=5008;NS=2504;DP=15021;EA...  GT:AS  ./.:1,0  ...   \n",
      "1  AC=578;AF=0.115415;AN=5008;NS=2504;DP=21675;EA...  GT:AS  ./.:0,0  ...   \n",
      "2  AC=593;AF=0.118411;AN=5008;NS=2504;DP=18233;EA...  GT:AS  0/0:4,0  ...   \n",
      "3  AC=588;AF=0.117412;AN=5008;NS=2504;DP=19730;EA...  GT:AS  0/0:3,0  ...   \n",
      "4  AC=594;AF=0.11861;AN=5008;NS=2504;DP=18859;EAS...  GT:AS  0/0:3,0  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:1,0  ./.:0,0  ./.:1,0  ./.:0,0  ./.:1,0  ./.:1,0  ./.:0,1  ./.:0,0   \n",
      "1  0/0:2,0  ./.:1,0  0/0:3,0  ./.:2,0  0/0:4,0  0/0:2,1  0/1:0,0  ./.:1,0   \n",
      "2  ./.:2,0  ./.:2,0  0/0:4,0  0/0:2,0  0/0:1,0  0/0:0,1  0/1:4,2  ./.:1,0   \n",
      "3  0/0:2,0  0/0:4,0  0/0:0,0  0/0:4,0  0/0:1,0  0/0:0,0  ./.:2,4  0/0:3,0   \n",
      "4  ./.:2,0  0/0:3,0  0/0:0,0  0/0:4,0  0/0:2,0  0/0:1,0  0/1:2,3  0/0:1,0   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:3,0  ./.:0,0  \n",
      "1  0/0:5,0  ./.:2,0  \n",
      "2  0/0:5,0  0/1:1,0  \n",
      "3  0/0:3,0  0/1:3,1  \n",
      "4  0/0:3,0  0/1:3,2  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                       0    1    2    3    4    5    6    7    8    9   ...  \\\n",
      "0  chr8_11844635_11845990  220   98  238  253  357  206  234  273  302  ...   \n",
      "1  chr8_11848319_11848530   42   32   36   55   31   35   32   25   38  ...   \n",
      "2  chr8_11853546_11854514  278  133  148  465  382  227  224  335  458  ...   \n",
      "3  chr8_16859214_16859438   15    2   40   26   12   22    5    9   22  ...   \n",
      "4  chr8_22599221_22600284  162   92  251  381  374  216  235  250  519  ...   \n",
      "\n",
      "    23   24   25   26   27  28   29   30   31   32  \n",
      "0  240  233  238  252  291  84  352  233  451  223  \n",
      "1   26   32   34   38   29   9   45   35   61   21  \n",
      "2  323  234  166  309  342  78  409  273  366  249  \n",
      "3   17   16   11   26   74   4   47   20   41   18  \n",
      "4  347  149  285  186  369  83  234  307  614  118  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  170\n",
      "Feature SNPs (m):  18\n",
      "chr9\n",
      "     0         1           2  3  4    5     6   \\\n",
      "0  chr9  17569343   rs2254582  C  T  100  PASS   \n",
      "1  chr9  17572071   rs2754341  C  T  100  PASS   \n",
      "2  chr9  17578504   rs3758217  C  T  100  PASS   \n",
      "3  chr9  17578709   rs3758216  G  C  100  PASS   \n",
      "4  chr9  17579692  rs13294100  T  G  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=2774;AF=0.553914;AN=5008;NS=2504;DP=18909;E...  GT:AS  0/1:0,0  ...   \n",
      "1  AC=3178;AF=0.634585;AN=5008;NS=2504;DP=19959;E...  GT:AS  ./.:0,1  ...   \n",
      "2  AC=2853;AF=0.569688;AN=5008;NS=2504;DP=18801;E...  GT:AS  0/1:1,1  ...   \n",
      "3  AC=3355;AF=0.669928;AN=5008;NS=2504;DP=17194;E...  GT:AS  0/1:2,0  ...   \n",
      "4  AC=2290;AF=0.457268;AN=5008;NS=2504;DP=10782;E...  GT:AS  0/1:2,3  ...   \n",
      "\n",
      "         31        32       33       34        35       36        37  \\\n",
      "0   ./.:0,0   ./.:0,0  ./.:0,1  ./.:1,0   ./.:0,3  ./.:0,1   ./.:2,0   \n",
      "1   ./.:0,1   ./.:0,0  ./.:0,1  ./.:0,0   ./.:0,2  ./.:2,0   ./.:0,0   \n",
      "2   1/1:0,4  1/1:0,10  0/1:2,4  0/1:1,1   1/1:0,4  1/1:3,5  0/0:12,0   \n",
      "3   1/1:0,9  1/1:0,15  0/1:4,4  0/1:5,4   1/1:0,8  1/1:2,5  0/0:11,0   \n",
      "4  1/1:0,10  1/1:0,10  0/1:4,6  0/1:7,5  1/1:0,18  1/1:6,4  0/0:25,0   \n",
      "\n",
      "         38       39       40  \n",
      "0   ./.:0,0  ./.:0,0  ./.:0,0  \n",
      "1   0/1:0,1  ./.:0,0  ./.:0,0  \n",
      "2   0/1:1,3  0/1:1,3  0/1:1,3  \n",
      "3  1/1:0,10  0/1:8,5  0/1:0,4  \n",
      "4   1/1:0,8  0/1:3,1  0/1:5,6  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                       0    1    2     3     4     5     6    7     8     9   \\\n",
      "0  chr9_17578173_17579882  376  153   338   507   814   336  372   459   923   \n",
      "1  chr9_17699014_17699637   86   53    80   117   128   100   79   104   131   \n",
      "2  chr9_17730813_17731948   40   35    75    46    74    51   39    66    56   \n",
      "3  chr9_33816144_33818416  791  310  1845  1857  1595  1628  956  1055  1975   \n",
      "4  chr9_33957284_33957573   76   28    30    76    86    57   56   104   108   \n",
      "\n",
      "   ...    23   24    25    26    27   28    29    30    31   32  \n",
      "0  ...   486  232   363   419   624  215   399   471   556  294  \n",
      "1  ...   126   75    61   120   134   34   158    91   103  108  \n",
      "2  ...    91   49    46    45    84   45   107    66    65   68  \n",
      "3  ...  1279  776  1419  1400  1640  469  1511  1508  2478  608  \n",
      "4  ...    90   44    39    81   108   16   105    69    72   70  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  81\n",
      "Feature SNPs (m):  6\n",
      "chr10\n",
      "      0         1           2  3  4    5     6   \\\n",
      "0  chr10  11665916  rs11257227  A  G  100  PASS   \n",
      "1  chr10  11672508  rs74347557  C  T  100  PASS   \n",
      "2  chr10  11672687  rs77892763  C  T  100  PASS   \n",
      "3  chr10  11675398  rs11257238  T  C  100  PASS   \n",
      "4  chr10  11676332   rs2011704  T  C  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=1245;AF=0.248602;AN=5008;NS=2504;DP=18322;E...  GT:AS  ./.:0,0  ...   \n",
      "1  AC=1304;AF=0.260383;AN=5008;NS=2504;DP=10867;E...  GT:AS  ./.:0,0  ...   \n",
      "2  AC=1321;AF=0.263778;AN=5008;NS=2504;DP=14662;E...  GT:AS  ./.:0,0  ...   \n",
      "3  AC=1281;AF=0.255791;AN=5008;NS=2504;DP=18710;E...  GT:AS  ./.:0,0  ...   \n",
      "4  AC=1324;AF=0.264377;AN=5008;NS=2504;DP=21542;E...  GT:AS  ./.:2,0  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:0,0  ./.:1,1  ./.:0,0  ./.:0,0  0/1:1,0  0/1:0,0  ./.:0,0  ./.:1,0   \n",
      "1  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:0,0   \n",
      "2  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,2  ./.:0,0  ./.:1,0  ./.:0,0  ./.:2,0   \n",
      "3  ./.:0,0  ./.:0,1  ./.:0,0  ./.:1,1  ./.:1,2  ./.:0,0  ./.:5,0  ./.:0,0   \n",
      "4  ./.:1,1  ./.:2,1  ./.:0,0  ./.:1,0  ./.:1,3  ./.:1,0  ./.:1,0  0/0:0,0   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:0,1  ./.:0,0  \n",
      "1  ./.:0,1  ./.:1,0  \n",
      "2  ./.:0,0  ./.:0,0  \n",
      "3  ./.:0,0  ./.:0,1  \n",
      "4  ./.:0,1  ./.:0,1  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                        0   1   2   3    4   5    6   7    8    9   ...   23  \\\n",
      "0  chr10_11678218_11678420  13   5  18   18  26   14   6   11   24  ...   20   \n",
      "1  chr10_15514705_15515858  64  21  93   87  65  131  47   56   70  ...   90   \n",
      "2  chr10_80490946_80491146   8   4  22    8   6    8   5   11   11  ...   10   \n",
      "3  chr10_80491172_80492195  91  54  68  120  97   67  20   78  105  ...  116   \n",
      "4  chr10_80493572_80494352  44  29  99   78  97   59  41  148  170  ...  100   \n",
      "\n",
      "   24   25  26   27  28   29   30   31   32  \n",
      "0  14   16  14   30   6   37   17   18   27  \n",
      "1  51  117  89  101  72  159  108   97   51  \n",
      "2   8   13   9    9  11   18    6   11   11  \n",
      "3  65  119  57  158  41  118   98  140   79  \n",
      "4  70  137  62  152  33  111   67  141  144  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  143\n",
      "Feature SNPs (m):  11\n",
      "chr11\n",
      "      0         1           2  3  4    5     6   \\\n",
      "0  chr11  10511509   rs4910149  C  A  100  PASS   \n",
      "1  chr11  10511626   rs4910150  C  T  100  PASS   \n",
      "2  chr11  10513802   rs7929890  G  A  100  PASS   \n",
      "3  chr11  10514751   rs1047730  T  C  100  PASS   \n",
      "4  chr11  10516034  rs73411920  G  T  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=1217;AF=0.243011;AN=5008;NS=2504;DP=19726;E...  GT:AS  0/1:0,2  ...   \n",
      "1  AC=1217;AF=0.243011;AN=5008;NS=2504;DP=18324;E...  GT:AS  0/1:1,0  ...   \n",
      "2  AC=1252;AF=0.25;AN=5008;NS=2504;DP=14660;EAS_A...  GT:AS  0/1:0,1  ...   \n",
      "3  AC=1217;AF=0.243011;AN=5008;NS=2504;DP=19440;E...  GT:AS  ./.:0,0  ...   \n",
      "4  AC=1207;AF=0.241014;AN=5008;NS=2504;DP=16389;E...  GT:AS  0/1:1,2  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  0/1:0,2  0/1:1,0  0/1:2,0  0/0:4,0  0/1:1,4  0/1:0,0  0/0:8,0  0/0:2,0   \n",
      "1  0/1:1,1  0/1:3,0  ./.:3,1  0/0:4,0  0/1:1,0  0/1:1,0  0/0:1,0  ./.:1,0   \n",
      "2  0/1:0,1  0/1:3,1  ./.:0,1  0/0:4,0  0/1:1,0  0/1:3,0  0/0:6,0  0/0:0,0   \n",
      "3  0/1:1,0  ./.:1,0  ./.:0,0  ./.:1,0  ./.:0,0  ./.:1,0  ./.:3,0  ./.:1,0   \n",
      "4  0/1:0,0  0/1:0,1  ./.:2,1  0/0:5,0  ./.:1,0  ./.:1,0  ./.:1,0  ./.:1,0   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:2,0  0/0:5,0  \n",
      "1  ./.:2,0  0/0:0,0  \n",
      "2  ./.:0,0  0/0:2,0  \n",
      "3  ./.:2,0  ./.:1,0  \n",
      "4  ./.:3,0  ./.:2,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                        0    1    2    3    4    5    6    7    8    9   ...  \\\n",
      "0  chr11_10510208_10511825  111   80  133  149  108  231  108  116  120  ...   \n",
      "1  chr11_10516461_10516661   22   20   29   26   24   51   25   16   24  ...   \n",
      "2  chr11_10522137_10523353  176  117  353  478  211  527  203  240  182  ...   \n",
      "3  chr11_47254627_47255399   65   53   70  146  131  101   66  122  150  ...   \n",
      "4  chr11_47272059_47272287   60   38   80   62   85   99   62   78   82  ...   \n",
      "\n",
      "    23   24   25   26   27  28   29   30   31   32  \n",
      "0  181   73  142  172  148  51  236  146  173  125  \n",
      "1   39   16   16   33   27  14   53   38   49   44  \n",
      "2  383  182  323  308  395  61  611  323  357  301  \n",
      "3  145   78   77  124  129  52  169  115  176  112  \n",
      "4   89   58   69   71   57  16   89   78  123   72  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  888\n",
      "Feature SNPs (m):  52\n",
      "chr12\n",
      "      0         1            2  3  4    5     6   \\\n",
      "0  chr12  32657825  rs181609621  A  C  100  PASS   \n",
      "1  chr12  32826971  rs148163066  A  G  100  PASS   \n",
      "2  chr12  33013238  rs150168440  T  C  100  PASS   \n",
      "3  chr12  33024607  rs140105005  C  T  100  PASS   \n",
      "4  chr12  33050376  rs117084165  A  G  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=12;AF=0.00239617;AN=5008;NS=2504;DP=16986;E...  GT:AS  0/0:5,0  ...   \n",
      "1  AC=13;AF=0.00259585;AN=5008;NS=2504;DP=19186;E...  GT:AS  ./.:1,0  ...   \n",
      "2  AC=14;AF=0.00279553;AN=5008;NS=2504;DP=18331;E...  GT:AS  ./.:1,0  ...   \n",
      "3  AC=14;AF=0.00279553;AN=5008;NS=2504;DP=19249;E...  GT:AS  ./.:0,0  ...   \n",
      "4  AC=15;AF=0.00299521;AN=5008;NS=2504;DP=18087;E...  GT:AS  ./.:2,0  ...   \n",
      "\n",
      "        31        32       33       34        35       36        37       38  \\\n",
      "0  0/0:7,0  0/0:12,0  0/0:4,0  0/0:3,0  0/0:22,0  0/0:6,0  0/0:11,0  0/0:8,0   \n",
      "1  ./.:1,0   ./.:1,0  ./.:1,0  ./.:1,0   ./.:0,0  ./.:1,0   ./.:2,0  ./.:0,2   \n",
      "2  ./.:0,0   ./.:0,0  ./.:1,0  ./.:1,0   ./.:2,0  ./.:2,0   ./.:1,0  ./.:0,1   \n",
      "3  ./.:3,0   ./.:0,0  ./.:0,0  ./.:0,0   ./.:1,0  ./.:1,0   ./.:1,0  ./.:1,0   \n",
      "4  ./.:1,0   ./.:2,0  ./.:0,0  ./.:3,0   ./.:2,0  ./.:2,0   ./.:0,0  ./.:0,2   \n",
      "\n",
      "        39       40  \n",
      "0  0/0:5,0  0/0:4,0  \n",
      "1  ./.:1,0  ./.:0,0  \n",
      "2  ./.:0,0  ./.:1,0  \n",
      "3  ./.:1,0  ./.:4,0  \n",
      "4  ./.:2,0  ./.:1,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                        0    1   2    3    4    5    6    7    8    9   ...  \\\n",
      "0  chr12_32656909_32658271  130  80  130  159  177  144  140  151  205  ...   \n",
      "1  chr12_33530318_33531622  192  83  102  271  363  186  201  307  346  ...   \n",
      "2  chr12_33567027_33568351  204  80  143  264  364  158  120  254  345  ...   \n",
      "3  chr12_39989928_39990795  111  62  112  114  251  146  129  174  215  ...   \n",
      "4  chr12_39990823_39991552   56  41   66   50   99   46   77   80  113  ...   \n",
      "\n",
      "    23   24   25   26   27   28   29   30   31   32  \n",
      "0  203  123  130  158  294   68  206  200  253  165  \n",
      "1  295  145  111  345  466  122  427  294  227  232  \n",
      "2  363   91  192  337  514  127  293  248  158  284  \n",
      "3  208  107  127  185  298  106  241  194  200  193  \n",
      "4  111   52   78   96  139   57  115   98  106  115  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  222\n",
      "Feature SNPs (m):  19\n",
      "chr13\n",
      "      0         1          2  3  4    5     6   \\\n",
      "0  chr13  49341255  rs6561520  T  C  100  PASS   \n",
      "1  chr13  49351229  rs7991335  A  C  100  PASS   \n",
      "2  chr13  49353596  rs9568188  T  C  100  PASS   \n",
      "3  chr13  49362741  rs7326978  A  G  100  PASS   \n",
      "4  chr13  49365767  rs9535211  C  T  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=1564;AF=0.3123;AN=5008;NS=2504;DP=17004;EAS...  GT:AS  ./.:0,0  ...   \n",
      "1  AC=2222;AF=0.44369;AN=5008;NS=2504;DP=17336;EA...  GT:AS  0/1:1,3  ...   \n",
      "2  AC=2312;AF=0.461661;AN=5008;NS=2504;DP=18607;E...  GT:AS  ./.:0,3  ...   \n",
      "3  AC=2167;AF=0.432708;AN=5008;NS=2504;DP=17846;E...  GT:AS  ./.:0,0  ...   \n",
      "4  AC=1465;AF=0.292532;AN=5008;NS=2504;DP=20415;E...  GT:AS  ./.:0,1  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,1   \n",
      "1  ./.:1,2  ./.:0,0  0/0:0,0  ./.:0,0  ./.:1,1  ./.:0,3  ./.:0,4  ./.:0,1   \n",
      "2  ./.:2,0  ./.:0,1  ./.:2,0  ./.:0,2  ./.:2,0  ./.:0,1  ./.:0,2  ./.:0,1   \n",
      "3  ./.:0,0  ./.:0,0  ./.:1,0  ./.:0,1  ./.:0,0  ./.:1,0  ./.:0,1  ./.:0,0   \n",
      "4  ./.:1,0  ./.:0,1  ./.:0,0  ./.:0,1  ./.:3,1  ./.:1,3  ./.:0,1  ./.:0,1   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:0,0  ./.:0,0  \n",
      "1  ./.:0,2  0/0:1,0  \n",
      "2  ./.:0,0  ./.:0,0  \n",
      "3  ./.:1,1  ./.:1,0  \n",
      "4  ./.:2,0  ./.:0,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                        0    1    2    3     4     5     6    7    8    9   \\\n",
      "0  chr13_97210544_97213057  640  324  985  1226  1130  1494  750  976  852   \n",
      "\n",
      "   ...    23   24   25    26    27   28    29    30    31   32  \n",
      "0  ...  1165  591  992  1205  1537  341  1568  1075  1211  999  \n",
      "\n",
      "[1 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  10\n",
      "Feature SNPs (m):  1\n",
      "chr14\n",
      "      0         1           2  3  4    5     6   \\\n",
      "0  chr14  22998455   rs3829409  G  A  100  PASS   \n",
      "1  chr14  23008384   rs8003201  G  A  100  PASS   \n",
      "2  chr14  37519905  rs12147939  T  C  100  PASS   \n",
      "3  chr14  37520065  rs12147950  T  C  100  PASS   \n",
      "4  chr14  37531976  rs17768343  A  C  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=408;AF=0.0814696;AN=5008;NS=2504;DP=16028;E...  GT:AS  0/0:0,0  ...   \n",
      "1  AC=863;AF=0.172324;AN=5008;NS=2504;DP=16889;EA...  GT:AS  0/0:0,0  ...   \n",
      "2  AC=3496;AF=0.698083;AN=5008;NS=2504;DP=16911;E...  GT:AS  ./.:0,0  ...   \n",
      "3  AC=2373;AF=0.473842;AN=5008;NS=2504;DP=18222;E...  GT:AS  ./.:0,0  ...   \n",
      "4  AC=2336;AF=0.466454;AN=5008;NS=2504;DP=16843;E...  GT:AS  0/1:1,1  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:0,0  0/0:2,0  0/0:0,0  0/0:3,0  0/0:0,0  0/0:2,0  0/1:2,0  ./.:1,0   \n",
      "1  ./.:0,0  0/0:6,0  ./.:1,0  0/0:6,0  0/0:2,0  0/0:3,0  0/1:6,5  0/0:4,0   \n",
      "2  ./.:0,0  ./.:0,1  ./.:0,0  ./.:0,0  ./.:0,0  ./.:2,0  ./.:1,0  ./.:1,0   \n",
      "3  ./.:0,0  ./.:0,0  ./.:0,1  ./.:1,1  ./.:0,0  ./.:0,0  ./.:0,1  ./.:2,0   \n",
      "4  ./.:0,0  ./.:0,0  0/1:0,0  0/1:0,1  0/1:1,2  0/1:2,0  ./.:4,1  ./.:1,0   \n",
      "\n",
      "        39       40  \n",
      "0  0/0:1,0  0/0:0,0  \n",
      "1  ./.:1,0  0/0:2,0  \n",
      "2  ./.:0,0  ./.:1,1  \n",
      "3  ./.:1,1  ./.:2,0  \n",
      "4  ./.:0,0  ./.:2,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                        0    1    2    3    4    5    6    7    8     9   ...  \\\n",
      "0  chr14_54881150_54882592  161  107  195  199  221  193  142  222   311  ...   \n",
      "1  chr14_54901731_54903475  503  237  480  771  793  479  449  586  1293  ...   \n",
      "2  chr14_67514560_67515927  301  153  327  465  653  331  265  335   794  ...   \n",
      "3  chr14_67516743_67517444   96   51  184  187  138  247  114  125   184  ...   \n",
      "4  chr14_67522119_67522945   81   67  177  218  192  279   93  111   150  ...   \n",
      "\n",
      "    23   24   25   26   27   28   29   30   31   32  \n",
      "0  240  137  231  189  418  105  266  154  286  258  \n",
      "1  773  349  518  520  874  249  520  709  931  338  \n",
      "2  432  232  368  294  477  136  322  408  710  145  \n",
      "3  234  139  147  189  305   64  330  140  321  121  \n",
      "4  213  116  191  189  308   57  305  146  385  116  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  154\n",
      "Feature SNPs (m):  9\n",
      "chr15\n",
      "      0         1           2  3  4    5     6   \\\n",
      "0  chr15  58680796  rs11854073  G  A  100  PASS   \n",
      "1  chr15  58682705   rs4775087  T  C  100  PASS   \n",
      "2  chr15  58690191  rs12439457  T  C  100  PASS   \n",
      "3  chr15  58691226   rs7161889  G  C  100  PASS   \n",
      "4  chr15  58691389   rs7163162  T  A  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=3110;AF=0.621006;AN=5008;NS=2504;DP=17901;E...  GT:AS  ./.:0,0  ...   \n",
      "1  AC=3135;AF=0.625998;AN=5008;NS=2504;DP=16089;E...  GT:AS  1/1:0,2  ...   \n",
      "2  AC=3137;AF=0.626398;AN=5008;NS=2504;DP=18179;E...  GT:AS  ./.:0,1  ...   \n",
      "3  AC=3137;AF=0.626398;AN=5008;NS=2504;DP=21790;E...  GT:AS  ./.:0,2  ...   \n",
      "4  AC=3137;AF=0.626398;AN=5008;NS=2504;DP=20270;E...  GT:AS  1/1:0,0  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  0/0:1,0  ./.:1,1  ./.:1,0  ./.:2,0  ./.:2,0  ./.:1,0  ./.:5,0  0/0:2,0   \n",
      "1  0/0:1,0  ./.:2,0  0/0:2,0  ./.:2,0  ./.:2,0  ./.:2,0  0/0:5,0  ./.:1,0   \n",
      "2  0/0:0,0  0/1:2,2  0/0:1,0  0/0:5,0  ./.:2,0  ./.:1,0  ./.:5,0  ./.:1,0   \n",
      "3  0/0:3,0  ./.:0,1  ./.:0,0  ./.:3,0  0/0:1,0  0/0:0,0  ./.:5,0  0/0:2,0   \n",
      "4  0/0:2,0  ./.:1,1  ./.:1,0  0/0:3,0  0/0:4,0  0/0:2,0  0/0:4,0  0/0:0,0   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:1,1  ./.:3,0  \n",
      "1  ./.:1,1  ./.:3,0  \n",
      "2  ./.:1,1  0/0:1,0  \n",
      "3  ./.:1,0  0/0:1,0  \n",
      "4  ./.:0,1  ./.:0,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                        0    1    2     3     4     5     6     7     8   \\\n",
      "0  chr15_58690546_58691615   68   31    79    91    85    97    59    73   \n",
      "1  chr15_58748498_58750792  861  476  2209  2067  1728  1968  1228  1376   \n",
      "2  chr15_58757822_58758463   30   36    62    85    52   111    40    56   \n",
      "3  chr15_58878599_58878799   14   15     9    27    27    15    13    17   \n",
      "4  chr15_63276586_63278773  574  282   787  1100   933   852   590   726   \n",
      "\n",
      "     9   ...    23   24    25    26    27   28    29    30    31   32  \n",
      "0    55  ...   110   68    84    98    79   56   112    72   123   68  \n",
      "1  2022  ...  1467  841  1812  1706  1684  334  1934  1630  2609  859  \n",
      "2    50  ...    66   32    56    91    68   24   139    65    96   41  \n",
      "3    29  ...    24   20    12    28    17   12    27    23    31   17  \n",
      "4  1176  ...   906  478   677   754   905  170   957   812  1157  506  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  95\n",
      "Feature SNPs (m):  9\n",
      "chr16\n",
      "      0         1           2  3  4    5     6   \\\n",
      "0  chr16  19266171   rs6497339  A  T  100  PASS   \n",
      "1  chr16  19268058   rs1051547  T  C  100  PASS   \n",
      "2  chr16  19268142     rs11343  T  G  100  PASS   \n",
      "3  chr16  28910488   rs7184597  T  C  100  PASS   \n",
      "4  chr16  28910828  rs11646653  C  T  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=3733;AF=0.745407;AN=5008;NS=2504;DP=20068;E...  GT:AS  ./.:0,0  ...   \n",
      "1  AC=3969;AF=0.792532;AN=5008;NS=2504;DP=22014;E...  GT:AS  ./.:0,1  ...   \n",
      "2  AC=3965;AF=0.791733;AN=5008;NS=2504;DP=19918;E...  GT:AS  ./.:0,0  ...   \n",
      "3  AC=4126;AF=0.823882;AN=5008;NS=2504;DP=16833;E...  GT:AS  ./.:0,0  ...   \n",
      "4  AC=3781;AF=0.754992;AN=5008;NS=2504;DP=17675;E...  GT:AS  ./.:0,1  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:3,0  ./.:0,1  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:0,0   \n",
      "1  ./.:0,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:0,1  ./.:0,1  ./.:0,1  ./.:0,1   \n",
      "2  ./.:0,0  ./.:0,2  ./.:0,0  ./.:0,1  ./.:0,0  ./.:0,1  ./.:1,0  ./.:0,0   \n",
      "3  ./.:0,0  ./.:1,0  ./.:0,0  ./.:0,3  ./.:0,0  ./.:0,4  ./.:0,1  ./.:1,0   \n",
      "4  ./.:0,0  ./.:0,0  ./.:0,1  ./.:0,0  ./.:0,0  ./.:0,1  1/1:0,4  ./.:0,1   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:0,0  ./.:0,0  \n",
      "1  ./.:0,0  ./.:3,0  \n",
      "2  ./.:1,0  ./.:0,0  \n",
      "3  ./.:0,2  ./.:0,0  \n",
      "4  ./.:0,1  ./.:0,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                        0    1    2    3    4    5    6    7    8    9   ...  \\\n",
      "0  chr16_19265219_19266531   45   35   71   95   66   38   52   65   90  ...   \n",
      "1  chr16_28925148_28926205  366  178  761  901  554  670  490  549  882  ...   \n",
      "2  chr16_30009486_30012051  351  225  355  557  507  302  298  457  622  ...   \n",
      "3  chr16_30018596_30018926   13    7   43    9   10    9    9   24   25  ...   \n",
      "4  chr16_30030716_30031966  171   78  176  319  248  190  175  234  331  ...   \n",
      "\n",
      "    23   24   25   26   27  28   29   30   31   32  \n",
      "0   76   52   46   49  105  22  107   63   76   71  \n",
      "1  515  369  571  533  471  83  619  557  951  278  \n",
      "2  498  282  263  282  261  93  347  459  682  208  \n",
      "3   18    7   14   24   11   5   30   24   14    9  \n",
      "4  283  212  152  148  120  51  265  240  427  114  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  289\n",
      "Feature SNPs (m):  38\n",
      "chr17\n",
      "      0        1            2  3  4    5     6   \\\n",
      "0  chr17  4824057  rs113124299  C  A  100  PASS   \n",
      "1  chr17  4826117   rs72835011  G  C  100  PASS   \n",
      "2  chr17  4838423    rs7225684  A  G  100  PASS   \n",
      "3  chr17  4839649   rs72835012  A  G  100  PASS   \n",
      "4  chr17  4840224   rs72835013  G  C  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=144;AF=0.028754;AN=5008;NS=2504;DP=10973;EA...  GT:AS  ./.:1,0  ...   \n",
      "1  AC=147;AF=0.029353;AN=5008;NS=2504;DP=16134;EA...  GT:AS  ./.:1,0  ...   \n",
      "2  AC=301;AF=0.0601038;AN=5008;NS=2504;DP=17681;E...  GT:AS  0/0:2,0  ...   \n",
      "3  AC=248;AF=0.0495208;AN=5008;NS=2504;DP=19338;E...  GT:AS  ./.:1,0  ...   \n",
      "4  AC=247;AF=0.0493211;AN=5008;NS=2504;DP=17391;E...  GT:AS  ./.:1,0  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:1,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:1,0  ./.:1,0  ./.:2,0  ./.:0,0   \n",
      "1  ./.:0,0  ./.:0,0  ./.:1,0  ./.:1,0  ./.:0,0  ./.:1,0  ./.:3,0  ./.:0,0   \n",
      "2  0/0:2,0  0/0:5,0  0/0:0,0  0/0:4,0  0/0:2,0  0/0:2,0  0/0:5,0  ./.:4,0   \n",
      "3  0/0:0,0  ./.:3,0  ./.:1,0  0/0:1,0  ./.:1,0  ./.:1,0  0/0:6,0  ./.:1,0   \n",
      "4  ./.:1,0  ./.:3,0  ./.:0,0  ./.:3,0  ./.:1,0  ./.:0,0  ./.:4,0  ./.:2,0   \n",
      "\n",
      "         39       40  \n",
      "0   ./.:1,0  ./.:0,0  \n",
      "1   ./.:1,0  ./.:2,0  \n",
      "2  0/0:10,0  0/0:3,0  \n",
      "3   ./.:0,0  ./.:0,0  \n",
      "4   ./.:2,0  ./.:0,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                      0    1    2     3     4    5    6    7    8    9   ...  \\\n",
      "0  chr17_4842175_4843328  214  122   137   306  369  184  170  320  361  ...   \n",
      "1  chr17_4850406_4850622   15    8    13    31   19   19    9   20   35  ...   \n",
      "2  chr17_4857485_4859285  267  142  1009  1026  380  995  357  396  338  ...   \n",
      "3  chr17_4859987_4860304   34   23    84    85   36  116   23   28   43  ...   \n",
      "4  chr17_4872615_4873120   43   32   118    93   82  135   53   69   47  ...   \n",
      "\n",
      "    23   24   25   26   27  28    29   30   31   32  \n",
      "0  338  230  156  237  213  64   381  214  338  224  \n",
      "1   35   16   12   21   29   7    43   10   35   21  \n",
      "2  544  357  687  708  361  93  1143  621  743  310  \n",
      "3   75   49   56   64   31  13   106   54  101   40  \n",
      "4   76   42  101  103   55  23   137   59  165   47  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  2930\n",
      "Feature SNPs (m):  152\n",
      "chr18\n",
      "      0         1           2  3  4    5     6   \\\n",
      "0  chr18  31495082  rs74481103  C  T  100  PASS   \n",
      "1  chr18  31499237  rs73414289  T  G  100  PASS   \n",
      "2  chr18  31500684  rs56848605  G  A  100  PASS   \n",
      "3  chr18  31501797  rs57047618  C  G  100  PASS   \n",
      "4  chr18  31502268  rs59665004  T  C  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=290;AF=0.0579073;AN=5008;NS=2504;DP=23104;E...  GT:AS  ./.:0,0  ...   \n",
      "1  AC=587;AF=0.117212;AN=5008;NS=2504;DP=16477;EA...  GT:AS  ./.:1,0  ...   \n",
      "2  AC=620;AF=0.123802;AN=5008;NS=2504;DP=19919;EA...  GT:AS  ./.:1,0  ...   \n",
      "3  AC=620;AF=0.123802;AN=5008;NS=2504;DP=17796;EA...  GT:AS  ./.:0,0  ...   \n",
      "4  AC=574;AF=0.114617;AN=5008;NS=2504;DP=12638;EA...  GT:AS  ./.:0,0  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:1,0  ./.:0,0  0/0:0,0  ./.:2,0  ./.:1,0  ./.:0,0  ./.:1,0  ./.:2,0   \n",
      "1  0/0:4,0  ./.:1,0  ./.:1,0  ./.:0,0  ./.:4,0  ./.:3,0  ./.:1,0  ./.:1,0   \n",
      "2  0/0:3,0  0/0:7,0  0/0:1,0  0/0:4,0  0/0:3,0  0/0:3,0  0/0:4,0  0/0:3,0   \n",
      "3  ./.:1,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:1,0  ./.:2,0  ./.:2,0  ./.:2,0   \n",
      "4  0/0:1,0  ./.:3,0  ./.:0,0  ./.:1,0  ./.:2,0  ./.:3,0  ./.:0,0  ./.:1,0   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:1,0  ./.:0,0  \n",
      "1  ./.:0,0  ./.:1,0  \n",
      "2  0/0:4,0  0/0:6,0  \n",
      "3  ./.:1,0  ./.:2,0  \n",
      "4  ./.:0,0  ./.:1,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                        0    1   2    3    4    5    6    7    8    9   ...  \\\n",
      "0  chr18_31500492_31500787   15  12   28   25   72   11   23   44   55  ...   \n",
      "1  chr18_31506437_31506651    5   7    7    8   18   12    7   10   12  ...   \n",
      "2  chr18_33744828_33746352  189  92  213  232  305  264  161  192  234  ...   \n",
      "3  chr18_33752235_33752479   37  15   14   32   45   44   20   23   21  ...   \n",
      "4  chr18_33820520_33821312   57  52   44   52   63   48   16   70  117  ...   \n",
      "\n",
      "    23   24   25   26   27   28   29   30   31   32  \n",
      "0   40   23   21   19   33   18   27   27   40   30  \n",
      "1   20   15   11   12   33   12   17   16   11   21  \n",
      "2  280  243  238  207  459  160  338  261  430  210  \n",
      "3   43   23   46   35   56   28   69   41   28   41  \n",
      "4   51   29   49   45   95   55   77   72   55   74  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  114\n",
      "Feature SNPs (m):  10\n",
      "chr19\n",
      "      0        1            2  3  4    5     6   \\\n",
      "0  chr19  1036692   rs11670349  C  T  100  PASS   \n",
      "1  chr19  1036705   rs11670373  C  T  100  PASS   \n",
      "2  chr19  1038572    rs1057874  C  T  100  PASS   \n",
      "3  chr19  1039187    rs4807468  G  A  100  PASS   \n",
      "4  chr19  1039324  rs111278892  C  G  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=367;AF=0.0732827;AN=5008;NS=2504;DP=13037;E...  GT:AS  ./.:0,0  ...   \n",
      "1  AC=370;AF=0.0738818;AN=5008;NS=2504;DP=13109;E...  GT:AS  ./.:0,0  ...   \n",
      "2  AC=3616;AF=0.722045;AN=5008;NS=2504;DP=16189;E...  GT:AS  ./.:0,2  ...   \n",
      "3  AC=3589;AF=0.716653;AN=5008;NS=2504;DP=16298;E...  GT:AS  ./.:0,2  ...   \n",
      "4  AC=357;AF=0.0712859;AN=5008;NS=2504;DP=16460;E...  GT:AS  0/0:2,0  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:0,1  ./.:2,0  ./.:0,0  ./.:2,0  ./.:0,0  ./.:0,0  ./.:0,1  ./.:0,0   \n",
      "1  ./.:0,0  ./.:2,0  ./.:1,0  ./.:2,0  ./.:0,0  ./.:0,0  ./.:0,1  ./.:0,0   \n",
      "2  0/1:0,1  ./.:0,2  ./.:0,1  1/1:0,0  0/1:1,3  0/1:0,0  0/0:6,0  1/1:0,2   \n",
      "3  0/1:1,1  ./.:0,3  ./.:0,1  ./.:0,4  ./.:0,0  ./.:0,2  0/0:7,0  ./.:0,2   \n",
      "4  0/1:3,2  ./.:3,0  ./.:3,0  0/0:3,0  0/0:4,0  0/0:2,0  ./.:0,3  ./.:0,0   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:0,0  ./.:0,0  \n",
      "1  ./.:0,0  ./.:0,0  \n",
      "2  ./.:2,2  ./.:0,2  \n",
      "3  ./.:1,0  ./.:0,1  \n",
      "4  0/0:7,0  ./.:1,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                        0    1    2    3     4    5    6    7    8     9   \\\n",
      "0    chr19_1039617_1041468  709  353  942  1229  990  952  704  795  1300   \n",
      "1  chr19_18416382_18416812  161   86  178   344  229  184  199  262   331   \n",
      "2  chr19_18428649_18429483  372  198  329   742  554  433  392  568   931   \n",
      "3  chr19_18438041_18438292  111   61   87   204  121   97   82  127   215   \n",
      "4  chr19_18453332_18453979   61   37  101    76   61   92   44   39    51   \n",
      "\n",
      "   ...    23   24   25   26   27   28    29   30    31   32  \n",
      "0  ...  1021  627  899  978  609  164  1265  988  1314  529  \n",
      "1  ...   239  156  182  199  186   27   260  195   237  124  \n",
      "2  ...   497  279  373  284  313   88   428  405   662  264  \n",
      "3  ...   129   67   89   51   66   18    81  115   158   54  \n",
      "4  ...    81   47   63   69   43   14    93   70    96   39  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  1134\n",
      "Feature SNPs (m):  58\n",
      "chr20\n",
      "      0        1           2  3  4    5     6   \\\n",
      "0  chr20  3170699  rs35420316  G  A  100  PASS   \n",
      "1  chr20  3172857  rs55785911  G  A  100  PASS   \n",
      "2  chr20  3174729   rs4813629  G  A  100  PASS   \n",
      "3  chr20  3180364  rs67820322  C  T  100  PASS   \n",
      "4  chr20  3182073  rs13045634  C  T  100  PASS   \n",
      "\n",
      "                                                  7      8         9   ...  \\\n",
      "0  AC=1962;AF=0.391773;AN=5008;NS=2504;DP=17251;E...  GT:AS   0/0:1,0  ...   \n",
      "1  AC=2002;AF=0.39976;AN=5008;NS=2504;DP=14736;EA...  GT:AS  0/0:14,0  ...   \n",
      "2  AC=2531;AF=0.505391;AN=5008;NS=2504;DP=17815;E...  GT:AS   0/1:1,1  ...   \n",
      "3  AC=2283;AF=0.455871;AN=5008;NS=2504;DP=16233;E...  GT:AS   0/1:1,0  ...   \n",
      "4  AC=2283;AF=0.455871;AN=5008;NS=2504;DP=8193;EA...  GT:AS   ./.:0,0  ...   \n",
      "\n",
      "        31        32       33        34       35       36        37        38  \\\n",
      "0  0/1:5,2   0/0:5,0  0/1:1,1   0/0:4,0  0/1:3,0  0/1:1,0   0/1:5,3   1/1:0,4   \n",
      "1  0/1:8,4  0/0:22,0  0/1:4,2  0/0:14,0  0/1:4,4  0/1:5,3   0/1:8,2  1/1:0,12   \n",
      "2  0/1:2,1   0/0:8,0  0/1:2,3   0/0:8,0  0/1:2,4  0/1:4,1  0/0:15,1   1/1:0,8   \n",
      "3  0/1:2,2   0/0:9,0  ./.:1,0   ./.:2,0  0/1:0,2  0/1:0,0   0/1:2,4   ./.:0,0   \n",
      "4  ./.:0,1   ./.:3,0  ./.:0,0   ./.:0,0  ./.:2,2  ./.:0,0   ./.:1,0   ./.:0,0   \n",
      "\n",
      "         39       40  \n",
      "0   0/1:3,2  0/1:5,2  \n",
      "1  0/1:10,8  0/1:5,5  \n",
      "2   ./.:0,5  0/1:7,2  \n",
      "3   0/1:1,0  ./.:1,2  \n",
      "4   ./.:0,0  ./.:0,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                        0    1    2    3     4     5    6    7    8     9   \\\n",
      "0    chr20_3172129_3173436  480  214  426   853   868  583  510  686   913   \n",
      "1    chr20_3189093_3189737  151   68   99   257   223  114  133  206   215   \n",
      "2    chr20_6005079_6006676  611  283  911  1052  1109  943  732  809  1425   \n",
      "3    chr20_6030325_6031410  199  139  109   235   272  155  172  295   369   \n",
      "4  chr20_56411726_56412255   18    8   75    62    46   58   31   38    45   \n",
      "\n",
      "   ...   23   24   25   26    27   28   29    30    31   32  \n",
      "0  ...  691  418  479  565   589  215  685   650   919  468  \n",
      "1  ...  225  100  107  166   191   20  233   178   166  185  \n",
      "2  ...  839  503  759  746  1099  346  952  1028  1285  440  \n",
      "3  ...  267  159  134  233   246   53  262   155   220  237  \n",
      "4  ...   54   40   55   63    75   20  104    25    93   52  \n",
      "\n",
      "[5 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  82\n",
      "Feature SNPs (m):  6\n",
      "chr21\n",
      "      0         1           2  3  4    5     6   \\\n",
      "0  chr21  37398228  rs11701722  T  C  100  PASS   \n",
      "1  chr21  37417654   rs2298328  T  C  100  PASS   \n",
      "2  chr21  37430571   rs2835736  C  G  100  PASS   \n",
      "3  chr21  37439246  rs73408672  T  G  100  PASS   \n",
      "4  chr21  37443528   rs2835746  A  G  100  PASS   \n",
      "\n",
      "                                                  7      8        9   ...  \\\n",
      "0  AC=1520;AF=0.303514;AN=5008;NS=2504;DP=9113;EA...  GT:AS  ./.:0,1  ...   \n",
      "1  AC=1936;AF=0.386581;AN=5008;NS=2504;DP=13840;E...  GT:AS  ./.:3,0  ...   \n",
      "2  AC=1827;AF=0.364816;AN=5008;NS=2504;DP=20285;E...  GT:AS  ./.:0,1  ...   \n",
      "3  AC=1574;AF=0.314297;AN=5008;NS=2504;DP=19805;E...  GT:AS  ./.:0,1  ...   \n",
      "4  AC=1915;AF=0.382388;AN=5008;NS=2504;DP=21087;E...  GT:AS  ./.:0,0  ...   \n",
      "\n",
      "        31       32       33       34       35       36       37       38  \\\n",
      "0  ./.:0,1  ./.:1,0  ./.:0,0  ./.:0,0  ./.:0,0  ./.:3,0  ./.:1,0  ./.:1,1   \n",
      "1  ./.:0,3  ./.:1,0  ./.:1,0  ./.:3,0  ./.:3,0  ./.:1,0  ./.:2,0  ./.:1,2   \n",
      "2  ./.:0,5  ./.:0,0  ./.:0,0  ./.:1,0  0/0:4,0  0/0:0,0  ./.:3,0  ./.:0,0   \n",
      "3  ./.:0,2  ./.:3,0  0/0:1,0  ./.:2,0  ./.:4,0  ./.:0,0  0/0:5,0  ./.:0,2   \n",
      "4  ./.:0,1  ./.:2,0  ./.:0,0  ./.:2,0  ./.:0,0  ./.:0,0  ./.:4,0  ./.:1,2   \n",
      "\n",
      "        39       40  \n",
      "0  ./.:0,0  ./.:0,0  \n",
      "1  ./.:0,0  ./.:0,2  \n",
      "2  ./.:0,0  0/1:2,0  \n",
      "3  ./.:2,0  ./.:0,1  \n",
      "4  ./.:1,0  ./.:0,0  \n",
      "\n",
      "[5 rows x 41 columns]\n",
      "                        0    1    2   3    4    5    6    7    8    9   ...  \\\n",
      "0  chr21_37451892_37452691  166  101  96  224  223  130  158  162  237  ...   \n",
      "1  chr21_37452714_37453157   41   40  37   87   82   50   50   65   67  ...   \n",
      "2  chr21_37485164_37485614   28   40  34   49   31   40   21   65   41  ...   \n",
      "\n",
      "    23  24   25   26   27  28   29   30   31   32  \n",
      "0  230  99  105  200  189  57  239  140  246  127  \n",
      "1   95  55   36   79   63  32   94   52   50   36  \n",
      "2   67  40   29   36   81  16   72   43   65   69  \n",
      "\n",
      "[3 rows x 33 columns]\n",
      "Samples (n):  33\n",
      "Testing SNPs (l):  34\n",
      "Feature SNPs (m):  3\n",
      "chr22\n"
     ]
    },
    {
     "ename": "EmptyDataError",
     "evalue": "No columns to parse from file",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mEmptyDataError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-29-09df32977aca>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      7\u001b[0m                             + reg + '/' + chrom + '/asvcf.gz', header=None, sep='\\t')\n\u001b[1;32m      8\u001b[0m         counts = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n\u001b[0;32m----> 9\u001b[0;31m                             + reg + '/' + chrom + '/Y.txt', header=None, sep='\\t')\n\u001b[0m\u001b[1;32m     10\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0masvcf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcounts\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhead\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36mparser_f\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, dialect, error_bad_lines, warn_bad_lines, delim_whitespace, low_memory, memory_map, float_precision)\u001b[0m\n\u001b[1;32m    683\u001b[0m         )\n\u001b[1;32m    684\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 685\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    686\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    687\u001b[0m     \u001b[0mparser_f\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    455\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    456\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 457\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfp_or_buf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    458\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    459\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m    893\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"has_index_names\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"has_index_names\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    894\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 895\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    896\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    897\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[0;34m(self, engine)\u001b[0m\n\u001b[1;32m   1133\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"c\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1134\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"c\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1135\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCParserWrapper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1136\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1137\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"python\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/miniconda3/lib/python3.7/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, src, **kwds)\u001b[0m\n\u001b[1;32m   1915\u001b[0m         \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"usecols\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0musecols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1916\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1917\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparsers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTextReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msrc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1918\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munnamed_cols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1919\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader.__cinit__\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mEmptyDataError\u001b[0m: No columns to parse from file"
     ]
    }
   ],
   "source": [
    "chroms_noXY = ['chr' + str(i) for i in range(1, 23)]\n",
    "for reg in multi_bam_regs:\n",
    "    print(reg)\n",
    "    for chrom in chroms_noXY:\n",
    "        print(chrom)\n",
    "        asvcf = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                            + reg + '/' + chrom + '/asvcf.gz', header=None, sep='\\t')\n",
    "        counts = pd.read_csv('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/'\n",
    "                            + reg + '/' + chrom + '/Y.txt', header=None, sep='\\t')\n",
    "        print(asvcf.head())\n",
    "        print(counts.head())\n",
    "        samples_n = counts.shape[1]\n",
    "        print(\"Samples (n): \", samples_n)\n",
    "        testing_snps_l = len(asvcf)\n",
    "        print(\"Testing SNPs (l): \", testing_snps_l)\n",
    "        feature_snps_m = len(counts)\n",
    "        print(\"Feature SNPs (m): \", feature_snps_m)\n",
    "        starts = ','.join([i.split('_')[1] for i in counts[0]])\n",
    "        #print(\"Starts: \", starts)\n",
    "        ends = ','.join([i.split('_')[2] for i in counts[0]])\n",
    "        #print(\"Ends: \", ends)\n",
    "        with open('/oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/scripts/'\n",
    "                     + reg + '/' + chrom + '/run_rasqual.sh', 'w') as outfile:\n",
    "            for feat in range(1, feature_snps_m + 1):\n",
    "                rasqual_cmd = 'zcat /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' \\\n",
    "                                + reg + '/' + chrom + '/asvcf.gz | /home/users/soumyak/rasqual/bin/rasqual -y /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' \\\n",
    "                                + reg + '/' + chrom + '/Y.bin -k /oak/stanford/groups/akundaje/projects/alzheimers_parkinsons/allelic_imbalance/rasqual/input/' \\\n",
    "                                + reg + '/' + chrom + '/K.bin -n ' + str(samples_n) + ' -j ' + str(feat) + ' -l ' + str(testing_snps_l) + ' -m ' \\\n",
    "                                + str(feature_snps_m) + ' -s ' + starts + ' -e ' + ends\n",
    "                outfile.write(rasqual_cmd + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
